--- mozilla/xpcom/build/Makefile.in.old	2008-04-28 20:20:37.160655000 +0800
+++ mozilla/xpcom/build/Makefile.in	2008-04-28 20:20:55.910870000 +0800
@@ -69,6 +69,7 @@
 
 GRE_MODULE	= 1
 MOZILLA_INTERNAL_API = 1
+DIST_INSTALL = 1
 
 REQUIRES	= string \
 		  $(NULL)
--- mozilla/browser/installer/Makefile.in.old	2008-04-28 20:16:37.328820000 +0800
+++ mozilla/browser/installer/Makefile.in	2008-04-28 20:17:01.871718000 +0800
@@ -108,3 +108,5 @@
 ifdef INSTALLER_DIR
 	$(MAKE) -C $(INSTALLER_DIR)
 endif
+
+libs:: make-sdk
--- mozilla/browser/app/Makefile.in.old	2008-04-28 20:17:21.914035000 +0800
+++ mozilla/browser/app/Makefile.in	2008-04-28 20:17:52.671996000 +0800
@@ -118,7 +118,7 @@
 endif
 
 ifdef MOZ_ENABLE_LIBXUL
-APP_XPCOM_LIBS = $(XPCOM_GLUE_LDOPTS)
+APP_XPCOM_LIBS = -lxpcom_core $(XPCOM_GLUE_LDOPTS)
 else
 MOZILLA_INTERNAL_API = 1
 APP_XPCOM_LIBS = $(XPCOM_LIBS)
@@ -126,11 +126,11 @@
 
 LIBS += \
 	$(STATIC_COMPONENTS_LINKER_PATH) \
-	$(EXTRA_DSO_LIBS) \
 	$(MOZ_JS_LIBS) \
 	$(APP_XPCOM_LIBS) \
 	$(NSPR_LIBS) \
 	$(TK_LIBS) \
+	$(EXTRA_DSO_LIBS) \
 	$(NULL)
 
 # Add explicit X11 dependency when building against X11 toolkits
--- mozilla/toolkit/library/Makefile.in.old	2008-04-29 12:02:58.281362000 +0800
+++ mozilla/toolkit/library/Makefile.in	2008-04-29 12:03:36.698434000 +0800
@@ -188,7 +188,8 @@
 export:: $(RDF_UTIL_SRC_CPPSRCS) $(INTL_UNICHARUTIL_UTIL_CPPSRCS)
 	$(INSTALL) $^ .
 
-EXTRA_DSO_LDOPTS += $(LIBS_DIR) $(EXTRA_DSO_LIBS)
+EXTRA_DSO_LDOPTS += $(LIBS_DIR) $(EXTRA_DSO_LIBS) -z allextract $(LIBXUL_DIST)/lib/$(LIB_PREFIX)xpcomglue.$(LIB_SUFFIX) -z defaultextract
+
 
 ifdef MOZ_ENABLE_LIBXUL
 include $(srcdir)/libxul-rules.mk
--- mozilla/configure.in.old	2008-04-28 20:19:03.352360000 +0800
+++ mozilla/configure.in	2008-04-28 20:19:28.354332000 +0800
@@ -946,7 +946,7 @@
 MOZ_JS_LIBS='-L$(LIBXUL_DIST)/bin -lmozjs'
 DYNAMIC_XPCOM_LIBS='-L$(LIBXUL_DIST)/bin -lxpcom -lxpcom_core'
 MOZ_FIX_LINK_PATHS='-Wl,-rpath-link,$(LIBXUL_DIST)/bin'
-XPCOM_FROZEN_LDOPTS='-L$(LIBXUL_DIST)/bin -lxpcom'
+XPCOM_FROZEN_LDOPTS='-L$(LIBXUL_DIST)/bin -znolazyload -lxpcom -zlazyload'
 LIBXUL_LIBS='$(XPCOM_FROZEN_LDOPTS) -lxul'
 XPCOM_GLUE_LDOPTS='$(LIBXUL_DIST)/lib/$(LIB_PREFIX)xpcomglue_s.$(LIB_SUFFIX) $(XPCOM_FROZEN_LDOPTS)'
 XPCOM_STANDALONE_GLUE_LDOPTS='$(LIBXUL_DIST)/lib/$(LIB_PREFIX)xpcomglue.$(LIB_SUFFIX)'
--- mozilla/configure.old	2008-04-28 20:19:39.464153000 +0800
+++ mozilla/configure	2008-04-28 20:20:01.671106000 +0800
@@ -5227,7 +5227,7 @@
 MOZ_JS_LIBS='-L$(LIBXUL_DIST)/bin -lmozjs'
 DYNAMIC_XPCOM_LIBS='-L$(LIBXUL_DIST)/bin -lxpcom -lxpcom_core'
 MOZ_FIX_LINK_PATHS='-Wl,-rpath-link,$(LIBXUL_DIST)/bin'
-XPCOM_FROZEN_LDOPTS='-L$(LIBXUL_DIST)/bin -lxpcom'
+XPCOM_FROZEN_LDOPTS='-L$(LIBXUL_DIST)/bin -znolazyload -lxpcom -zlazyload'
 LIBXUL_LIBS='$(XPCOM_FROZEN_LDOPTS) -lxul'
 XPCOM_GLUE_LDOPTS='$(LIBXUL_DIST)/lib/$(LIB_PREFIX)xpcomglue_s.$(LIB_SUFFIX) $(XPCOM_FROZEN_LDOPTS)'
 XPCOM_STANDALONE_GLUE_LDOPTS='$(LIBXUL_DIST)/lib/$(LIB_PREFIX)xpcomglue.$(LIB_SUFFIX)'
