--- firefox/build/unix/mozilla-config.in.ori	2007-12-26 16:14:39.697272000 +0800
+++ firefox/build/unix/mozilla-config.in	2007-12-26 16:13:49.953215000 +0800
@@ -20,7 +20,6 @@
     *
 Libraries:
     xpcom
-    nspr
     js
     jsj
     gfx
@@ -74,7 +73,7 @@
     --idlflags)
       echo_idlflags=yes
       ;;
-    xpcom|js|nspr|gfx|jsj)
+    xpcom|js|gfx|jsj)
       echo_components="$echo_components $1"
       echo_libraries="$echo_libraries $1"
       ;;
@@ -107,15 +106,14 @@
     for n in $echo_components; do
 	component_includes="$component_includes -I%includedir%/$n"
     done
-    echo $component_includes $includes $nspr_cflags
+    echo $component_includes $includes
 fi
 
 if test "$echo_idlflags" = "yes"; then
     echo "-I%idldir%"
 fi
 
-_nspr_libs="%FULL_NSPR_LIBS%"
-_xpcom_libs="-lxpcom $_nspr_libs"
+_xpcom_libs="-lxpcom"
 _js_libs="-ljs"
 
 if test "$echo_libs" = "yes"; then
@@ -127,9 +125,6 @@
 	xpcom)
 	    libs="$libs $_xpcom_libs"
 	    ;;
-	nspr)
-	    libs="$libs $_nspr_libs"
-	    ;;
 	js)
 	    libs="$libs $_js_libs"
 	    ;;
--- firefox/toolkit/mozapps/installer/packager.mk.ori	2007-12-26 16:21:34.738674000 +0800
+++ firefox/toolkit/mozapps/installer/packager.mk	2007-12-26 15:57:42.839901000 +0800
@@ -211,6 +211,16 @@
 endif # !CROSS_COMPILE
 
 NO_PKG_FILES += \
+        libnspr4.so \
+        libplc4.so \
+        libplds4.so \
+        libnss3.so \
+        libsmime3.so \
+        libsoftokn3.chk \
+        libsoftokn3.so \
+        libssl3.so \
+        libnssckbi.so \
+        cpu/sparcv8plus/libnspr_flt4.so \
 	core \
 	bsdecho \
 	gtscc \
