--- mozilla/toolkit/xre/nsAppRunner.cpp.old6	2008-02-25 11:55:44.496494000 +0800
+++ mozilla/toolkit/xre/nsAppRunner.cpp	2008-02-25 11:57:32.869693000 +0800
@@ -2788,8 +2788,6 @@
       _gtk_window_set_auto_startup_notification(PR_FALSE);
     }
 
-    gtk_widget_set_default_visual(gdk_rgb_get_visual());
-    gtk_widget_set_default_colormap(gdk_rgb_get_cmap());
 #endif /* MOZ_WIDGET_GTK2 */
 
     // Call the code to install our handler
