--- kdenetwork-3.5.8/kopete/protocols/groupwise/libgroupwise/rtf.cc.orig	2008-02-24 13:54:52.353760039 +0530
+++ kdenetwork-3.5.8/kopete/protocols/groupwise/libgroupwise/rtf.cc	2008-02-24 13:55:24.215780549 +0530
@@ -2298,7 +2298,7 @@
 const unsigned GREEN		= 3;
 const unsigned BLUE			= 4;
 const unsigned CF			= 5;
-const unsigned FS			= 6;
+const unsigned FS_			= 6;
 const unsigned HIGHLIGHT	= 7;
 const unsigned PARD			= 8;
 const unsigned PAR			= 9;
@@ -2462,7 +2462,7 @@
                 case CF:
                     cur_level.setFontColor(cmd_value);
                     break;
-                case FS:
+                case FS_:
                     cur_level.setFontSizeHalfPoints(cmd_value);
                     break;
                 case HIGHLIGHT:
--- kdenetwork-3.5.8/kopete/protocols/groupwise/libgroupwise/rtf.ll.orig	2008-02-24 13:54:47.472260223 +0530
+++ kdenetwork-3.5.8/kopete/protocols/groupwise/libgroupwise/rtf.ll	2008-02-24 13:55:08.990948304 +0530
@@ -633,7 +633,7 @@
 const unsigned GREEN		= 3;
 const unsigned BLUE			= 4;
 const unsigned CF			= 5;
-const unsigned FS			= 6;
+const unsigned FS_			= 6;
 const unsigned HIGHLIGHT	= 7;
 const unsigned PARD			= 8;
 const unsigned PAR			= 9;
@@ -797,7 +797,7 @@
                 case CF:
                     cur_level.setFontColor(cmd_value);
                     break;
-                case FS:
+                case FS_:
                     cur_level.setFontSizeHalfPoints(cmd_value);
                     break;
                 case HIGHLIGHT:
