diff -ur libxfce4menu-orig/libxfce4menu/libxfce4menu-config.h.in libxfce4menu/libxfce4menu/libxfce4menu-config.h.in
--- libxfce4menu-orig/libxfce4menu/libxfce4menu-config.h.in	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/libxfce4menu-config.h.in	2007-04-07 20:45:56.590579072 +0700
@@ -32,7 +32,7 @@
 
 #include <glib-object.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #define LIBXFCE4MENU_MAJOR_VERSION @LIBXFCE4MENU_VERSION_MAJOR@
 #define LIBXFCE4MENU_MINOR_VERSION @LIBXFCE4MENU_VERSION_MINOR@
@@ -54,6 +54,6 @@
                                          guint required_minor,
                                          guint required_micro);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__LIBXFCE4MENU_CONFIG_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-and-rules.h libxfce4menu/libxfce4menu/xfce-menu-and-rules.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-and-rules.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-and-rules.h	2007-04-07 20:37:07.163818335 +0700
@@ -26,8 +26,11 @@
 #ifndef __XFCE_MENU_AND_RULES_H__
 #define __XFCE_MENU_AND_RULES_H__
 
+
 #include <glib-object.h>
 
+G_BEGIN_DECLS
+
 typedef struct _XfceMenuAndRules        XfceMenuAndRules;
 typedef struct _XfceMenuAndRulesClass   XfceMenuAndRulesClass;
 
@@ -38,10 +41,10 @@
 #define XFCE_IS_MENU_AND_RULES_CLASS(klass)  (G_TYPE_CHECK_CLASS_TYPE ((obj), XFCE_TYPE_MENU_AND_RULES))
 #define XFCE_MENU_AND_RULES_GET_CLASS(obj)   (G_TYPE_INSTANCE_GET_CLASS ((obj), XFCE_TYPE_MENU_AND_RULES, XfceMenuAndRulesClass))
 
-GType             xfce_menu_and_rules_get_type (void) G_GNUC_CONST;
+G_GNUC_CONST GType             xfce_menu_and_rules_get_type (void);
 
 XfceMenuAndRules *xfce_menu_and_rules_new      (void);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_AND_RULES_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-directory.h libxfce4menu/libxfce4menu/xfce-menu-directory.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-directory.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-directory.h	2007-04-07 20:29:51.659041115 +0700
@@ -28,7 +28,7 @@
 
 #include <glib-object.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceMenuDirectoryPrivate XfceMenuDirectoryPrivate;
 typedef struct _XfceMenuDirectoryClass   XfceMenuDirectoryClass;
@@ -41,7 +41,8 @@
 #define XFCE_IS_MENU_DIRECTORY_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), XFCE_TYPE_MENU_DIRECTORY))
 #define XFCE_MENU_DIRECTORY_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS ((obj), XFCE_TYPE_MENU_DIRECTORY, XfceMenuDirectoryClass))
 
-GType                    xfce_menu_directory_get_type            (void) G_GNUC_CONST;
+G_GNUC_CONST GType                    xfce_menu_directory_get_type            (void);
+
 
 const gchar             *xfce_menu_directory_get_filename        (XfceMenuDirectory *directory);
 void                     xfce_menu_directory_set_filename        (XfceMenuDirectory *directory,
@@ -62,10 +63,10 @@
 gboolean                 xfce_menu_directory_show_in_environment (XfceMenuDirectory *directory);
 
 #if defined(LIBXFCE4MENU_COMPILATION)
-void _xfce_menu_directory_init     (void) G_GNUC_INTERNAL;
-void _xfce_menu_directory_shutdown (void) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL void _xfce_menu_directory_init     (void);
+G_GNUC_INTERNAL void _xfce_menu_directory_shutdown (void);
 #endif
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_DIRECTORY_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-element.h libxfce4menu/libxfce4menu/xfce-menu-element.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-element.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-element.h	2007-04-07 20:30:17.507671307 +0700
@@ -29,7 +29,7 @@
 
 #include <glib-object.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceMenuElement      XfceMenuElement;
 typedef struct _XfceMenuElementIface XfceMenuElementIface;
@@ -48,11 +48,11 @@
   const gchar *(*get_icon_name) (XfceMenuElement *element);
 };
 
-GType        xfce_menu_element_get_type      (void) G_GNUC_CONST;
+G_GNUC_CONST GType        xfce_menu_element_get_type      (void);
 
 const gchar *xfce_menu_element_get_name      (XfceMenuElement *element);
 const gchar *xfce_menu_element_get_icon_name (XfceMenuElement *element);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_ELEMENT_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-item-cache.h libxfce4menu/libxfce4menu/xfce-menu-item-cache.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-item-cache.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-item-cache.h	2007-04-07 20:31:22.653000605 +0700
@@ -28,7 +28,7 @@
 
 #include <glib-object.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceMenuItemCachePrivate XfceMenuItemCachePrivate;
 typedef struct _XfceMenuItemCacheClass   XfceMenuItemCacheClass;
@@ -41,7 +41,7 @@
 #define XFCE_IS_MENU_ITEM_CACHE_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), XFCE_TYPE_MENU_ITEM_CACHE))
 #define XFCE_MENU_ITEM_CACHE_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS ((obj), XFCE_TYPE_MENU_ITEM_CACHE, XfceMenuItemCacheClass))
 
-GType              xfce_menu_item_cache_get_type    (void) G_GNUC_CONST;
+G_GNUC_CONST GType              xfce_menu_item_cache_get_type    (void);
 
 XfceMenuItemCache *xfce_menu_item_cache_get_default (void);
 
@@ -53,10 +53,10 @@
                                                      gpointer           user_data);
 
 #if defined(LIBXFCE4MENU_COMPILATION)
-void               _xfce_menu_item_cache_init       (void) G_GNUC_INTERNAL;
-void               _xfce_menu_item_cache_shutdown   (void) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL void               _xfce_menu_item_cache_init       (void);
+G_GNUC_INTERNAL void               _xfce_menu_item_cache_shutdown   (void);
 #endif
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_ITEM_CACHE_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-item-pool.h libxfce4menu/libxfce4menu/xfce-menu-item-pool.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-item-pool.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-item-pool.h	2007-04-07 20:31:43.693400995 +0700
@@ -29,7 +29,7 @@
 #include <glib-object.h>
 #include <libxfce4menu/xfce-menu-standard-rules.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceMenuItemPoolPrivate XfceMenuItemPoolPrivate;
 typedef struct _XfceMenuItemPoolClass   XfceMenuItemPoolClass;
@@ -42,7 +42,7 @@
 #define XFCE_IS_MENU_ITEM_POOL_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), XFCE_TYPE_MENU_ITEM_POOL))
 #define XFCE_MENU_ITEM_POOL_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS ((obj), XFCE_TYPE_MENU_ITEM_POOL, XfceMenuItemPoolClass))
 
-GType             xfce_menu_item_pool_get_type           (void) G_GNUC_CONST;
+G_GNUC_CONST GType             xfce_menu_item_pool_get_type           (void);
 
 XfceMenuItemPool *xfce_menu_item_pool_new                (void);
 
@@ -57,6 +57,6 @@
                                                           XfceMenuStandardRules *rule);
 gboolean          xfce_menu_item_pool_get_empty          (XfceMenuItemPool      *pool);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_ITEM_POOL_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-item.h libxfce4menu/libxfce4menu/xfce-menu-item.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-item.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-item.h	2007-04-07 20:32:05.109866922 +0700
@@ -29,7 +29,7 @@
 
 #include <glib-object.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceMenuItemPrivate XfceMenuItemPrivate;
 typedef struct _XfceMenuItemClass   XfceMenuItemClass;
@@ -42,7 +42,7 @@
 #define XFCE_IS_MENU_ITEM_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), XFCE_TYPE_MENU_ITEM))
 #define XFCE_MENU_ITEM_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS ((obj), XFCE_TYPE_MENU_ITEM, XfceMenuItemClass))
 
-GType         xfce_menu_item_get_type                          (void) G_GNUC_CONST;
+G_GNUC_CONST GType         xfce_menu_item_get_type                          (void);
 
 XfceMenuItem *xfce_menu_item_new                               (const gchar  *filename);
 
@@ -84,6 +84,6 @@
 void          xfce_menu_item_increment_allocated               (XfceMenuItem *item);
 void          xfce_menu_item_decrement_allocated               (XfceMenuItem *item);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_ITEM_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-layout.h libxfce4menu/libxfce4menu/xfce-menu-layout.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-layout.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-layout.h	2007-04-07 20:32:40.326609728 +0700
@@ -32,7 +32,7 @@
 
 #include <glib-object.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef enum
 {
@@ -63,9 +63,9 @@
 #define XFCE_IS_MENU_LAYOUT_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), XFCE_TYPE_MENU_LAYOUT))
 #define XFCE_MENU_LAYOUT_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS ((obj), XFCE_TYPE_MENU_LAYOUT, XfceMenuLayoutClass))
 
-GType                   xfce_menu_layout_get_type            (void) G_GNUC_CONST;
+G_GNUC_CONST GType                   xfce_menu_layout_get_type            (void);
 
-XfceMenuLayout         *xfce_menu_layout_new                 (void) G_GNUC_MALLOC;
+G_GNUC_MALLOC XfceMenuLayout         *xfce_menu_layout_new                 (void);
 void                    xfce_menu_layout_add_filename        (XfceMenuLayout          *layout,
                                                               const gchar             *filename);
 void                    xfce_menu_layout_add_menuname        (XfceMenuLayout          *layout,
@@ -84,7 +84,7 @@
 const gchar            *xfce_menu_layout_node_get_menuname   (XfceMenuLayoutNode       *node);
 XfceMenuLayoutMergeType xfce_menu_layout_node_get_merge_type (XfceMenuLayoutNode       *node);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_LAYOUT_H__ */
 
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-monitor.h libxfce4menu/libxfce4menu/xfce-menu-monitor.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-monitor.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-monitor.h	2007-04-07 20:33:06.942930470 +0700
@@ -30,7 +30,7 @@
 #include <glib.h>
 #include <libxfce4menu/libxfce4menu.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceMenuMonitorVTable XfceMenuMonitorVTable;
 
@@ -69,10 +69,10 @@
 };
 
 #if defined(LIBXFCE4MENU_COMPILATION)
-void _xfce_menu_monitor_init     (void) G_GNUC_INTERNAL;
-void _xfce_menu_monitor_shutdown (void) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL void _xfce_menu_monitor_init     (void);
+G_GNUC_INTERNAL void _xfce_menu_monitor_shutdown (void);
 #endif
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_MONITOR_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-move.h libxfce4menu/libxfce4menu/xfce-menu-move.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-move.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-move.h	2007-04-07 20:33:31.239500998 +0700
@@ -28,7 +28,7 @@
 
 #include <glib-object.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceMenuMoveClass XfceMenuMoveClass;
 typedef struct _XfceMenuMove      XfceMenuMove;
@@ -40,7 +40,7 @@
 #define XFCE_IS_MENU_MOVE_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), XFCE_TYPE_MENU_MOVE))
 #define XFCE_MENU_MOVE_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS ((obj), XFCE_TYPE_MENU_MOVE, XfceMenuMoveClass))
 
-GType         xfce_menu_move_get_type    (void) G_GNUC_CONST;
+G_GNUC_CONST GType         xfce_menu_move_get_type    (void);
 
 XfceMenuMove *xfce_menu_move_new         (void);
 
@@ -52,6 +52,6 @@
 void          xfce_menu_move_set_new     (XfceMenuMove *move,
                                           const gchar  *new);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_MOVE_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-not-rules.h libxfce4menu/libxfce4menu/xfce-menu-not-rules.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-not-rules.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-not-rules.h	2007-04-07 20:34:02.168232179 +0700
@@ -28,6 +28,8 @@
 
 #include <glib-object.h>
 
+G_BEGIN_DECLS
+
 typedef struct _XfceMenuNotRules        XfceMenuNotRules;
 typedef struct _XfceMenuNotRulesClass   XfceMenuNotRulesClass;
 
@@ -38,10 +40,10 @@
 #define XFCE_IS_MENU_NOT_RULES_CLASS(klass)  (G_TYPE_CHECK_CLASS_TYPE ((obj), XFCE_TYPE_MENU_NOT_RULES))
 #define XFCE_MENU_NOT_RULES_GET_CLASS(obj)   (G_TYPE_INSTANCE_GET_CLASS ((obj), XFCE_TYPE_MENU_NOT_RULES, XfceMenuNotRulesClass))
 
-GType             xfce_menu_not_rules_get_type (void) G_GNUC_CONST;
+G_GNUC_CONST GType             xfce_menu_not_rules_get_type (void);
 
 XfceMenuNotRules *xfce_menu_not_rules_new      (void);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_NOT_RULES_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-or-rules.h libxfce4menu/libxfce4menu/xfce-menu-or-rules.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-or-rules.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-or-rules.h	2007-04-07 20:34:35.824819652 +0700
@@ -28,6 +28,8 @@
 
 #include <glib-object.h>
 
+G_BEGIN_DECLS
+
 typedef struct _XfceMenuOrRules        XfceMenuOrRules;
 typedef struct _XfceMenuOrRulesClass   XfceMenuOrRulesClass;
 
@@ -38,10 +40,10 @@
 #define XFCE_IS_MENU_OR_RULES_CLASS(klass)  (G_TYPE_CHECK_CLASS_TYPE ((obj), XFCE_TYPE_MENU_OR_RULES))
 #define XFCE_MENU_OR_RULES_GET_CLASS(obj)   (G_TYPE_INSTANCE_GET_CLASS ((obj), XFCE_TYPE_MENU_OR_RULES, XfceMenuOrRulesClass))
 
-GType            xfce_menu_or_rules_get_type (void) G_GNUC_CONST;
+G_GNUC_CONST GType            xfce_menu_or_rules_get_type (void);
 
 XfceMenuOrRules *xfce_menu_or_rules_new      (void);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_OR_RULES_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-rules.h libxfce4menu/libxfce4menu/xfce-menu-rules.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-rules.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-rules.h	2007-04-07 20:34:53.649188081 +0700
@@ -29,7 +29,7 @@
 
 #include <glib-object.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceMenuRules      XfceMenuRules;
 typedef struct _XfceMenuRulesIface XfceMenuRulesIface;
@@ -56,7 +56,7 @@
 
 };
 
-GType    xfce_menu_rules_get_type    (void) G_GNUC_CONST;
+G_GNUC_CONST GType    xfce_menu_rules_get_type    (void);
 
 gboolean xfce_menu_rules_match        (XfceMenuRules *rules,
                                        XfceMenuItem  *item);
@@ -68,6 +68,6 @@
 void     xfce_menu_rules_add_category (XfceMenuRules *rules,
                                        const gchar   *category);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_RULES_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-separator.h libxfce4menu/libxfce4menu/xfce-menu-separator.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-separator.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-separator.h	2007-04-07 20:35:33.274081427 +0700
@@ -28,7 +28,7 @@
 
 #include <glib-object.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceMenuSeparatorClass XfceMenuSeparatorClass;
 typedef struct _XfceMenuSeparator      XfceMenuSeparator;
@@ -41,15 +41,15 @@
 #define XFCE_MENU_SEPARATOR_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS ((obj), XFCE_TYPE_MENU_SEPARATOR, XfceMenuSeparatorClass))
 
 
-GType              xfce_menu_separator_get_type    (void) G_GNUC_CONST;
+G_GNUC_CONST GType              xfce_menu_separator_get_type    (void);
 
 XfceMenuSeparator *xfce_menu_separator_get_default (void);
 
 #if defined(LIBXFCE4MENU_COMPILATION)
-void               _xfce_menu_separator_init       (void) G_GNUC_INTERNAL;
-void               _xfce_menu_separator_shutdown   (void) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL void               _xfce_menu_separator_init       (void);
+G_GNUC_INTERNAL void               _xfce_menu_separator_shutdown   (void);
 #endif
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_SEPARATOR_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu-standard-rules.h libxfce4menu/libxfce4menu/xfce-menu-standard-rules.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu-standard-rules.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu-standard-rules.h	2007-04-07 20:35:52.410506638 +0700
@@ -66,12 +66,12 @@
   gboolean include;
 };
 
-GType    xfce_menu_standard_rules_get_type      (void) G_GNUC_CONST;
+G_GNUC_CONST GType    xfce_menu_standard_rules_get_type      (void);
 
 gboolean xfce_menu_standard_rules_get_include   (XfceMenuStandardRules *rules);
 void     xfce_menu_standard_rules_set_include   (XfceMenuStandardRules *rules,
                                                  gboolean               include);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_STANDARD_RULES_H__ */
diff -ur libxfce4menu-orig/libxfce4menu/xfce-menu.h libxfce4menu/libxfce4menu/xfce-menu.h
--- libxfce4menu-orig/libxfce4menu/xfce-menu.h	2007-04-07 09:33:48.000000000 +0700
+++ libxfce4menu/libxfce4menu/xfce-menu.h	2007-04-07 20:36:51.499577082 +0700
@@ -28,7 +28,7 @@
 
 #include <glib-object.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceMenuPrivate XfceMenuPrivate;
 typedef struct _XfceMenuClass   XfceMenuClass;
@@ -44,12 +44,14 @@
 void               xfce_menu_init                  (const gchar *env);
 void               xfce_menu_shutdown              (void);
 
-GType              xfce_menu_get_type              (void) G_GNUC_CONST;
+G_GNUC_CONST GType              xfce_menu_get_type              (void);
 
-XfceMenu          *xfce_menu_get_root              (GError           **error) G_GNUC_CONST;
+G_GNUC_CONST
+XfceMenu          *xfce_menu_get_root              (GError           **error);
 
+G_GNUC_MALLOC
 XfceMenu          *xfce_menu_new                   (const gchar       *filename,
-                                                    GError           **error) G_GNUC_MALLOC;
+                                                    GError           **error);
 
 const gchar       *xfce_menu_get_filename          (XfceMenu          *menu);
 void               xfce_menu_set_filename          (XfceMenu          *menu,
@@ -80,6 +82,6 @@
 gboolean           xfce_menu_has_layout            (XfceMenu          *menu);
 GSList            *xfce_menu_get_layout_elements   (XfceMenu          *menu);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_MENU_H__ */
