--- firefox/browser/app/mozilla.in.ori1	2007-12-26 15:32:31.006560000 +0800
+++ firefox/browser/app/mozilla.in	2007-12-26 15:37:55.199528000 +0800
@@ -131,6 +131,13 @@
   esac
 done
 
+## add Mozilla plugins directory in MOZ_PLUGIN_PATH
+## to make sure Firefox can pick up plugins there
+test -n "$MOZ_PLUGIN_PATH" && \
+  MOZ_PLUGIN_PATH=$MOZ_PLUGIN_PATH:/usr/lib/firefox/plugins:/usr/sfw/lib/mozilla/plugins || \
+  MOZ_PLUGIN_PATH=/usr/lib/firefox/plugins:/usr/sfw/lib/mozilla/plugins
+export MOZ_PLUGIN_PATH
+
 if [ $debugging = 1 ]
 then
   echo $dist_bin/run-mozilla.sh $script_args $dist_bin/$MOZILLA_BIN "$@"
