diff -up kdebase-workspace-4.3.0/plasma/shells/desktop/desktopcorona.cpp.default-applets kdebase-workspace-4.3.0/plasma/shells/desktop/desktopcorona.cpp
--- kdebase-workspace-4.3.0/plasma/shells/desktop/desktopcorona.cpp.default-applets	2009-06-17 15:07:42.000000000 -0500
+++ kdebase-workspace-4.3.0/plasma/shells/desktop/desktopcorona.cpp	2009-08-19 18:18:03.571937055 -0500
@@ -259,15 +259,18 @@ void DesktopCorona::loadDefaultLayout()
     loadDefaultApplet("tasks", panel);
     loadDefaultApplet("systemtray", panel);
 
-    Plasma::DataEngineManager *engines = Plasma::DataEngineManager::self();
-    Plasma::DataEngine *power = engines->loadEngine("powermanagement");
-    if (power) {
-        const QStringList &batteries = power->query("Battery")["sources"].toStringList();
-        if (!batteries.isEmpty()) {
-            loadDefaultApplet("battery", panel);
-        }
+    if ( (!QFile::exists("/usr/share/autostart/guidance-power-manager.desktop")) && 
+         (!QFile::exists("/usr/share/autostart/kpowersave-autostart.desktop")) ) {
+       Plasma::DataEngineManager *engines = Plasma::DataEngineManager::self();
+       Plasma::DataEngine *power = engines->loadEngine("powermanagement");
+       if (power) {
+          const QStringList &batteries = power->query("Battery")["sources"].toStringList();
+          if (!batteries.isEmpty()) {
+             loadDefaultApplet("battery", panel);
+          }
+       }
+       engines->unloadEngine("powermanagement");
     }
-    engines->unloadEngine("powermanagement");
 
     loadDefaultApplet("digital-clock", panel);
     emit containmentAdded(panel);
