--- apps/kinfocenter/pci/kpci_private.h.orig	2008-04-18 09:21:35.000000000 +0200
+++ apps/kinfocenter/pci/kpci_private.h	2008-08-06 12:11:23.411975289 +0200
@@ -43,7 +43,7 @@
 	QString name;
 };
 
-union pciInfo{
+union pciInfo {
 	unsigned char raw[256];
 	struct {
 		unsigned short vendor;
@@ -63,8 +63,8 @@
 				unsigned comFastBack:1;
 				unsigned comInterrupt:1;
 				unsigned comUnk:5;
-			} KDE_PACKED;
-		} KDE_PACKED;
+			} command_bits KDE_PACKED;
+		} command KDE_PACKED;
 		union {
 			unsigned short status;
 			struct {
@@ -81,8 +81,8 @@
 				unsigned statRecMasterAbort:1;
 				unsigned statSigSystemError:1;
 				unsigned statDetectedParity:1;
-			} KDE_PACKED;
-		} KDE_PACKED;
+			} status_bits KDE_PACKED;
+		} status KDE_PACKED;
 		unsigned char revision;
 		union {
 			unsigned char devProgIface;
@@ -93,8 +93,8 @@
 				unsigned progSecProgInd:1;
 				unsigned progUnk:3;
 				unsigned progIdeMaster:1;
-			} KDE_PACKED;
-		} KDE_PACKED;
+			} devProgIface_bits KDE_PACKED;
+		} devProgIface KDE_PACKED;
 		unsigned char devSubClass;
 		unsigned char devClass;
 		unsigned char cacheLineSize;
@@ -104,8 +104,8 @@
 			struct {
 				unsigned headerType:7;
 				unsigned multifunctional:1;
-			} KDE_PACKED;
-		} KDE_PACKED;
+			} headerType_bits KDE_PACKED;
+		} headerType KDE_PACKED;
 		union {
 			unsigned char bist;
 			struct {
@@ -113,8 +113,8 @@
 				unsigned bistUnk:2;
 				unsigned bistStart:1;
 				unsigned bistCapable:1;
-			} KDE_PACKED;
-		} KDE_PACKED;
+			} bist_bits KDE_PACKED;
+		} bist KDE_PACKED;
 		union {
 			struct { //header0
 				union {
@@ -124,7 +124,7 @@
 						unsigned baseAddressType:2;
 						unsigned baseAddressPref:1;
 						unsigned baseAddressAddr:28;
-					} KDE_PACKED;
+					} baseAddress_bits KDE_PACKED;
 				} KDE_PACKED mapping[6];
 				unsigned cardbusCis;
 				unsigned short subVendor;
@@ -135,16 +135,16 @@
 						unsigned romEnabled:1;
 						unsigned romUnk:10;
 						unsigned romAddr:21;
-					} KDE_PACKED;
-				} KDE_PACKED;
+					} romAddress_bits KDE_PACKED;
+				} romAddress KDE_PACKED;
 				unsigned char capabilityList;
 				unsigned char reserved1[7];
 				unsigned char interruptLine;
 				unsigned char interruptPin;
 				unsigned char minGnt;
 				unsigned char maxLat;
-			} KDE_PACKED;
-			struct{ //header1
+			} header0 KDE_PACKED;
+			struct { //header1
 				unsigned baseAddress0_2;
 				unsigned baseAddress1_2;
 				unsigned char primaryBus;
@@ -157,8 +157,8 @@
 						unsigned ioBaseType:1;
 						unsigned ioBaseUnk:3;
 						unsigned ioBaseAddr:4;
-					} KDE_PACKED;
-				} KDE_PACKED;
+					} ioBase_bits KDE_PACKED;
+				} ioBase KDE_PACKED;
 				unsigned char ioLimit;
 				union {
 					unsigned short secStatus;
@@ -176,16 +176,16 @@
 						unsigned secStatRecMasterAbort:1;
 						unsigned secStatSigSystemError:1;
 						unsigned secStatDetectedParity:1;
-					} KDE_PACKED;
-				} KDE_PACKED;
+					} secStatus_bits KDE_PACKED;
+				} secStatus KDE_PACKED;
 				union {
 					unsigned short memoryBase;
 					struct {						
 						unsigned memoryType:1;
 						unsigned memoryUnk:3;
 						unsigned memoryAddr:4;
-					} KDE_PACKED;
-				} KDE_PACKED;
+					} memoryBase_bits KDE_PACKED;
+				} memoryBase KDE_PACKED;
 				unsigned short memoryLimit;
 				union {
 					unsigned short prefMemoryBase;
@@ -193,8 +193,8 @@
 						unsigned prefMemoryType:1;
 						unsigned prefMemoryUnk:3;
 						unsigned prefMemoryAddr:4;
-					} KDE_PACKED;
-				} KDE_PACKED;
+					} prefMemoryBase_bits KDE_PACKED;
+				} prefMemoryBase KDE_PACKED;
 				unsigned short prefMemoryLimit;
 				unsigned prefBaseUpper32;
 				unsigned prefLimitUpper32;
@@ -221,9 +221,9 @@
 						unsigned bridgeControlDisTimStat:1;
 						unsigned bridgeControlDisTimeSerr:1;
 						unsigned bridgeControlUnk2:4;
-					} KDE_PACKED;
-				} KDE_PACKED;
-			} KDE_PACKED;
+					} bridgeControl_bits KDE_PACKED;
+				} bridgeControl KDE_PACKED;
+			} header1 KDE_PACKED;
 			struct{ //header2
 				unsigned baseAddress0_3;
 				union {
@@ -242,8 +242,8 @@
 						unsigned cbSecStatRecMasterAbort:1;
 						unsigned cbSecStatSigSystemError:1;
 						unsigned cbSecStatDetectedParity:1;
-					} KDE_PACKED;
-				} KDE_PACKED;
+					} cbSecStatus_bits KDE_PACKED;
+				} cbSecStatus KDE_PACKED;
 				unsigned char cbPrimaryBus;
 				unsigned char cbCardBus;
 				unsigned char cbSubordinateBus;
@@ -259,8 +259,8 @@
 							unsigned cbIoBaseType:1;
 							unsigned coIoBaseUnk:1;
 							unsigned cbIoBaseAddr:30;
-						} KDE_PACKED;
-					} KDE_PACKED;
+						} cbIoBase_bits KDE_PACKED;
+					} cbIoBase KDE_PACKED;
 					unsigned cbIoLimit;
 				} KDE_PACKED cbIo[2];
 				unsigned char interruptLine_3;
@@ -280,14 +280,14 @@
 						unsigned cbControlPref1:1;
 						unsigned cbControlPostWrites:1;
 						unsigned cbControlUnk2:5;
-					} KDE_PACKED;
-				} KDE_PACKED;
+					} cbControl_bits KDE_PACKED;
+				} cbControl KDE_PACKED;
 				unsigned short cbSubVendor;
 				unsigned short cbSubDevice;
 				unsigned short cbLegacyModeBase;
-			} KDE_PACKED;
-		} KDE_PACKED;
-	} KDE_PACKED;
+			} header2 KDE_PACKED;
+		} header KDE_PACKED;
+	} cooked KDE_PACKED;
 } KDE_PACKED;
 
 union agpInfo{
@@ -297,13 +297,13 @@
 			unsigned revMinor:4;
 			unsigned revMaior:4;
 			unsigned char unk;
-		} KDE_PACKED;
+		} revision KDE_PACKED;
 		union {
 			unsigned status;
 			struct {
 				unsigned statusEnhRate:4;
 				unsigned statusUnk:28;
-			} KDE_PACKED;
+			} status_bits0 KDE_PACKED;
 			struct {
 				unsigned statusRate:3;
 				unsigned statusMode:1;
@@ -319,14 +319,14 @@
 				unsigned statusIsochSupp:1;
 				unsigned statusUnk2:6;
 				unsigned statusReq:8;
-			} KDE_PACKED;
-		} KDE_PACKED;
+			} status_bits1 KDE_PACKED;
+		} status KDE_PACKED;
 		union {
 			unsigned config;
 			struct {
 				unsigned configEnhRate:4;
 				unsigned configUnk:28;
-			} KDE_PACKED;
+			} config_bits0 KDE_PACKED;
 			struct {
 				unsigned configRate:3;
 				unsigned configUnk1:1;
@@ -340,9 +340,9 @@
 				unsigned configOptReqSize:3;
 				unsigned configUnk3:8;
 				unsigned configReq:8;
-			} KDE_PACKED;
-		} KDE_PACKED;
-	} KDE_PACKED;
+			} config_bits1 KDE_PACKED;
+		} config KDE_PACKED;
+	} cooked KDE_PACKED;
 } KDE_PACKED;
 
 union pmInfo {
@@ -363,8 +363,8 @@
 				unsigned capsPmeD2:1;
 				unsigned capsPmeD3hot:1;
 				unsigned capsPmeD3cold:1;
-			} KDE_PACKED;
-		} KDE_PACKED;
+			} caps_bits KDE_PACKED;
+		} caps KDE_PACKED;
 		union {
 			unsigned short status;
 			struct {
@@ -374,18 +374,18 @@
 				unsigned statDataSelect:4;
 				unsigned statDataScale:2;
 				unsigned statPmeStat:1;
-			} KDE_PACKED;
-		} KDE_PACKED;
+			} status_bits KDE_PACKED;
+		} status KDE_PACKED;
 		union {
 			unsigned char statusBridge;
 			struct {
 				unsigned statBridgeUnk:6;
 				unsigned statBridgeBx:1;
 				unsigned statBridgeClock:1;
-			} KDE_PACKED;
-		} KDE_PACKED;
+			} statusBridge_bits KDE_PACKED;
+		} statusBridge KDE_PACKED;
 		unsigned char data;
-	} KDE_PACKED;
+	} cooked KDE_PACKED;
 } KDE_PACKED;
 
 union vpdInfo {
@@ -394,7 +394,7 @@
 		unsigned vpdAddress:15;
 		unsigned vpdTransfer:1;
 		unsigned vpdData;
-	} KDE_PACKED;
+	} cooked  KDE_PACKED;
 } KDE_PACKED;
 
 union msiInfo {
@@ -409,8 +409,8 @@
 				unsigned msi64bit:1;
 				unsigned msiPerVector:1;
 				unsigned msiUnk0:7;
-			} KDE_PACKED;
-		} KDE_PACKED;
+			} msiControl_bits KDE_PACKED;
+		} msiControl KDE_PACKED;
 		unsigned msiAddress;
 		union {
 			struct {
@@ -419,15 +419,15 @@
 				unsigned short msiUnk64;
 				unsigned msiMask64;
 				unsigned msiPending64;
-			} KDE_PACKED;
+			} address KDE_PACKED;
 			struct {
 				unsigned short msiData;
 				unsigned short msiUnk;			
 				unsigned msiMask;
 				unsigned msiPending;
-			} KDE_PACKED;
-		} KDE_PACKED;
-	} KDE_PACKED;
+			} data KDE_PACKED;
+		} payload KDE_PACKED;
+	} cooked KDE_PACKED;
 } KDE_PACKED;
 
 //In following arrays -1 mean default value
