--- kdebase-workspace-4.2.4/startkde.cmake.orig	2009-06-14 11:01:10.345472323 +0530
+++ kdebase-workspace-4.2.4/startkde.cmake	2009-06-14 11:04:57.756390934 +0530
@@ -1,8 +1,18 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #  DEFAULT KDE STARTUP SCRIPT ( @KDE_VERSION_STRING@ )
 #
 
+# Make sure we can find the binaries.
+PATH=@CMAKE_INSTALL_PREFIX@/qt4/bin:${PATH}
+export PATH
+
+# The .UTF-8 language settings cause iiimx to come up; this causes
+# plasma to crash - vanish - all the time with bad X_CopyArea
+# requests. So remove that setting because we have our own means
+# to change input methods.
+test -n "$LANG" && LANG=`basename "$LANG" ".UTF-8"`
+
 # When the X server dies we get a HUP signal from xinit. We must ignore it
 # because we still need to do some cleanup.
 trap 'echo GOT SIGHUP' HUP
