diff -ur kdebase-workspace-4.2.4/klipper/urlgrabber.cpp kdebase-workspace-4.2.4-kde#165726-klipper-crash/klipper/urlgrabber.cpp
--- kdebase-workspace-4.2.4/klipper/urlgrabber.cpp	2008-11-04 18:12:28.000000000 +0100
+++ kdebase-workspace-4.2.4-kde#165726-klipper-crash/klipper/urlgrabber.cpp	2009-06-05 01:56:23.000000000 +0200
@@ -153,7 +153,8 @@
     if ( m_myClipData.isEmpty() )
         return;
 
-    ActionListIterator it( matchingActions( m_myClipData ) );
+    ActionList matchingActionsList = matchingActions( m_myClipData );
+    ActionListIterator it( matchingActionsList );
     ClipAction *action = 0L;
     ClipCommand *command = 0L;
 
