--- kdegraphics-4.3.1/libs/libkdcraw/libraw/libraw/libraw.h.orig	2009-02-04 23:47:36.000000000 +0530
+++ kdegraphics-4.3.1/libs/libkdcraw/libraw/libraw/libraw.h	2009-09-06 23:43:34.292417113 +0530
@@ -29,6 +29,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 
+#if defined(WIN32) || defined(__sun)
+#define bzero(p,sz) memset(p,0,sz)
+#endif
 
 #include "libraw_datastream.h"
 #include "libraw_types.h"
