--- kdelibs-4.2.4/kdeui/util/kpixmapcache.cpp.orig	2009-06-04 20:52:00.037453997 +0530
+++ kdelibs-4.2.4/kdeui/util/kpixmapcache.cpp	2009-06-04 20:50:33.184172133 +0530
@@ -472,7 +472,7 @@
     }
     info->indexHeader = reinterpret_cast<KPixmapCacheIndexHeader *>(indexMem);
 #ifdef HAVE_MADVISE
-    madvise(indexMem, info->size, MADV_WILLNEED);
+    madvise((char *)indexMem, info->size, MADV_WILLNEED);
 #endif
 
     info->file->close();
