--- kdelibs-4.3.0/kdecore/localization/klocale.cpp.orig	2009-07-08 20:11:53.000000000 +0530
+++ kdelibs-4.3.0/kdecore/localization/klocale.cpp	2009-08-29 13:40:53.462777693 +0530
@@ -2240,10 +2240,16 @@
   systemLocale = "UTF-8";
 #endif
 
-  if (systemLocale == "ANSI_X3.4-1968") // means ascii, "C"; QTextCodec doesn't know, so avoid warning
+  QTextCodec* codec;
+  if (systemLocale == "ANSI_X3.4-1968") {// means ascii, "C"; QTextCodec doesn't know, so avoid warning
     systemLocale = "ISO-8859-1";
+  }
+
+  codec = QTextCodec::codecForName(systemLocale);
+  if ( ! codec ) {
+    codec = QTextCodec::codecForLocale();
+  }
 
-  QTextCodec* codec = QTextCodec::codecForName(systemLocale);
   if ( codec )
     setEncoding( codec->mibEnum() );
 #else
