--- nepomuk/rcgen/CMakeLists.txt.orig	2009-06-23 16:46:18.000000000 +0530
+++ nepomuk/rcgen/CMakeLists.txt	2009-09-05 09:27:38.703585721 +0530
@@ -18,9 +18,9 @@
 qt4_add_resources(nepomuk_rcgen_SRCS templates.qrc)
 kde4_add_executable(nepomuk-rcgen ${nepomuk_rcgen_SRCS})
 target_link_libraries(nepomuk-rcgen
-  ${SOPRANO_LIBRARIES}
   ${QT_QTCORE_LIBRARY}
   ${QT_QTDBUS_LIBRARY}
+  ${SOPRANO_LIBRARIES}
 )
 
 install(TARGETS nepomuk-rcgen ${INSTALL_TARGETS_DEFAULT_ARGS} )
