--- kdebindings-4.2.4/python/pykde4/sip/kdecore/typedefs.sip.orig	2009-06-14 19:25:12.648750683 +0530
+++ kdebindings-4.2.4/python/pykde4/sip/kdecore/typedefs.sip	2009-06-14 19:28:11.698127586 +0530
@@ -244,7 +244,8 @@
         return PyInstance_Check(sipPy);
 
     int iserr = 0;
-    TYPE *cpp = (TYPE *)sipForceConvertTo_TYPE (sipPy, &iserr);
+    TYPE *cpp = (TYPE *)sipForceConvertToType(sipPy, sipType_TYPE, sipTransferObj, \
+        SIP_NO_CONVERTORS, NULL, &iserr);
 
     if (iserr)
     {
