--- telepathy-qt.SVN-R2109.orig/src/Core/CMakeLists.txt	2008-05-14 23:43:21.859691000 -0400
+++ telepathy-qt.SVN-R2109/src/Core/CMakeLists.txt	2008-05-15 13:35:58.567536000 -0400
@@ -1,10 +1,29 @@
-set(QT_DONT_USE_QTGUI "YES")
+set(QT_DONT_USE_QTGUI TRUE)
 include(${QT_USE_FILE})
 
 set(INC_DIR ${CMAKE_SOURCE_DIR}/include)
 link_directories(${CMAKE_BINARY_DIR/src/Common})
 include_directories(${INC_DIR} ${QT_QTCORE_INCLUDE_DIR} ${QTDBUS_CFLAGS})
 
+FIND_LIBRARY(QT_QTCORE_LIBRARY
+    NAMES QtCore
+    PATHS ${QT_LIBRARY_DIR})
+
+FIND_LIBRARY(QT_QTDBUS_LIBRARY
+    NAMES QtDBus
+    PATHS ${QT_LIBRARY_DIR})
+
+IF(QT_QTCORE_LIBRARY AND QT_QTDBUS_LIBRARY)
+    SET(QT_LIBRARIES_FOUND TRUE)
+    MESSAGE(STATUS "Required QT Libraries found.")
+ELSE(QT_QTCORE_LIBRARY AND QT_QTDBUS_LIBRARY)
+    SET(QT_LIBRARIES_FOUND FALSE)
+    MESSAGE(FATAL_ERROR "Required QT Libraries NOT found.")
+ENDIF(QT_QTCORE_LIBRARY AND QT_QTDBUS_LIBRARY)
+
+MESSAGE(STATUS "QT_QTCORE_LIBRARY: ${QT_QTCORE_LIBRARY}")
+MESSAGE(STATUS "QT_QTDBUS_LIBRARY: ${QT_QTDBUS_LIBRARY}")
+
 # ######### QtTelepathyCore library #########
 
 set(library_SRCS
@@ -58,6 +77,6 @@
 qt4_wrap_cpp(library_MOC_SRCS ${library_MOC_HDRS})
 add_library(QtTelepathyCore SHARED ${library_SRCS} ${library_MOC_SRCS})
 set_target_properties(QtTelepathyCore PROPERTIES VERSION 0.1.0 SOVERSION 0)
-target_link_libraries(QtTelepathyCore QtTelepathyCommon ${QT_LIBRARIES} ${QDBUS_LDFLAGS})
+target_link_libraries(QtTelepathyCore QtTelepathyCommon ${QT_LIBRARIES} ${QT_QTCORE_LIBRARY} ${QT_QTDBUS_LIBRARY} ${QDBUS_LDFLAGS})
 install(TARGETS QtTelepathyCore DESTINATION ${LIB_INSTALL_DIR})
 
