--- usr/src/cmd/mdb/common/modules/genunix/memory.c.orig	Tue Jul  1 18:14:43 2008
+++ usr/src/cmd/mdb/common/modules/genunix/memory.c	Tue Jul  1 18:15:11 2008
@@ -397,7 +397,7 @@
 	}
 #endif
 
-	mdb_printf("Free (freelist)  %16llu  %16llu  %3llu%%\n", freemem,
+	mdb_printf("Free (freelist)  %16lu  %16llu  %3llu%%\n", freemem,
 	    (uint64_t)freemem * pagesize / (1024 * 1024),
 	    MS_PCT_TOTAL(freemem));
 
