Index: apcupsd-3.14.5/src/apcaccess.c
===================================================================
--- apcupsd-3.14.5.orig/src/apcaccess.c
+++ apcupsd-3.14.5/src/apcaccess.c
@@ -89,7 +89,7 @@ int main(int argc, char **argv)
       char *p;
 
       host = argv[2];
-      p = strchr(host, ':');
+      p = (char*)strchr(host, ':');
       if (p) {
          *p++ = 0;
          port = atoi(p);
