--- capseo_0.3.0/examples/encode-raw.c.orig	2009-06-08 12:35:18.000000000 +0530
+++ capseo_0.3.0/examples/encode-raw.c	2009-06-07 11:25:21.282848784 +0530
@@ -23,6 +23,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <math.h>
+#include <libgen.h>
 
 #include <GL/glx.h>
 
@@ -81,11 +82,18 @@
 }//}}}
 
 int main(int argc, char *argv[]) {
+
+        if (argc != 2) {
+            (void) fprintf(stderr, "Usage: %s <output-file-name>\n",
+                    basename(argv[0]));
+            return 1;
+        }
+
 	Display *dpy = XOpenDisplay(NULL);
 	Window win = createWindow(dpy, 400, 400);
 
 	capseo_info_t info;
-	bzero(&info, sizeof(info));
+	(void) memset(&info, '\0', sizeof(info));
 
 	info.mode = CAPSEO_MODE_ENCODE;
 	info.width = 400;
@@ -101,8 +109,8 @@
 		return die(strerror(errno));
 
 	struct { uint32_t width, height; } header = { 
-		htonl((long)(info.width / pow(2, info.scale))), 
-		htonl((long)(info.height / pow(2, info.scale)))
+		htonl((long)(info.width / pow(double(2), info.scale))), 
+		htonl((long)(info.height / pow(double(2), info.scale)))
 	};
 	write(fd, &header, sizeof(header));
 
