--- libarchive-2.6.2-64/libarchive/archive_write_disk.c.orig	2009-07-02 21:21:16.716636248 +0530
+++ libarchive-2.6.2-64/libarchive/archive_write_disk.c	2009-07-02 21:21:38.898445378 +0530
@@ -50,6 +50,7 @@
 #endif
 
 #if defined(HAVE_EXT2FS_EXT2_FS_H) && !defined(__CYGWIN__)
+#include <sys/ioccom.h>
 #include <ext2fs/ext2_fs.h>	/* Linux file flags, broken on Cygwin */
 #endif
 #ifdef HAVE_ERRNO_H
