diff -ur libxfcegui4-orig/libxfcegui4/icons.h libxfcegui4/libxfcegui4/icons.h
--- libxfcegui4-orig/libxfcegui4/icons.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/icons.h	2007-04-07 22:28:50.333841187 +0700
@@ -81,17 +81,24 @@
 
 
 #ifndef XFCE_DISABLE_DEPRECATED
-void xfce_themed_icon_add_search_path(const gchar *path) G_GNUC_DEPRECATED;
-void xfce_set_icon_theme(const gchar *theme_name) G_GNUC_DEPRECATED;
-GdkPixbuf *inline_icon_at_size(const guint8 *data,
-                               int width,
-                               int height) G_GNUC_DEPRECATED;
-GdkPixbuf *xfce_load_themed_icon(const gchar *name,
-                                 gint size) G_GNUC_DEPRECATED;
-GdkPixbuf *xfce_pixbuf_new_from_file_at_size(const gchar *filename,
-                                             gint width,
-                                             gint height,
-                                             GError **error) G_GNUC_DEPRECATED;
+G_GNUC_DEPRECATED void xfce_themed_icon_add_search_path(const gchar *path);
+
+G_GNUC_DEPRECATED void xfce_set_icon_theme(const gchar *theme_name);
+
+G_GNUC_DEPRECATED GdkPixbuf *
+inline_icon_at_size(const guint8 *data,
+                    int width,
+                    int height);
+
+G_GNUC_DEPRECATED GdkPixbuf *
+xfce_load_themed_icon(const gchar *name,
+                      gint size);
+
+G_GNUC_DEPRECATED GdkPixbuf *
+xfce_pixbuf_new_from_file_at_size(const gchar *filename,
+                                  gint width,
+                                  gint height,
+                                  GError **error);
 #endif
 
 G_END_DECLS
diff -ur libxfcegui4-orig/libxfcegui4/libxfcegui4-config.h.in libxfcegui4/libxfcegui4/libxfcegui4-config.h.in
--- libxfcegui4-orig/libxfcegui4/libxfcegui4-config.h.in	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/libxfcegui4-config.h.in	2007-04-07 22:28:50.334058383 +0700
@@ -28,7 +28,7 @@
 
 #include <libxfce4util/libxfce4util.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #define LIBXFCEGUI4_MAJOR_VERSION	@LIBXFCEGUI4_VERSION_MAJOR@
 #define LIBXFCEGUI4_MINOR_VERSION	@LIBXFCEGUI4_VERSION_MINOR@
@@ -42,6 +42,6 @@
        && LIBXFCEGUI4_MINOR_VERSION == (minor) \
        && LIBXFCEGUI4_MICRO_VERSION >= (micro)))
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif  /* !__LIBXFCEGUI4_LIBXFCEGUI4_CONFIG_H__ */
diff -ur libxfcegui4-orig/libxfcegui4/netk-application.h libxfcegui4/libxfcegui4/netk-application.h
--- libxfcegui4-orig/libxfcegui4/netk-application.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/netk-application.h	2007-04-07 22:28:50.334215968 +0700
@@ -55,9 +55,8 @@
     void (*icon_changed) (NetkApplication * app);
 };
 
-GType
-netk_application_get_type (void)
-    G_GNUC_CONST;
+G_GNUC_CONST GType
+netk_application_get_type (void);
 
      NetkApplication *netk_application_get (gulong xwindow);
 
diff -ur libxfcegui4-orig/libxfcegui4/netk-class-group.h libxfcegui4/libxfcegui4/netk-class-group.h
--- libxfcegui4-orig/libxfcegui4/netk-class-group.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/netk-class-group.h	2007-04-07 22:28:50.334347511 +0700
@@ -52,7 +52,7 @@
     void (*icon_changed) (NetkClassGroup * group);
 };
 
-GType netk_class_group_get_type (void) G_GNUC_CONST;
+G_GNUC_CONST GType netk_class_group_get_type (void);
 
 NetkClassGroup *netk_class_group_get (const char *res_class);
 
diff -ur libxfcegui4-orig/libxfcegui4/netk-pager.h libxfcegui4/libxfcegui4/netk-pager.h
--- libxfcegui4-orig/libxfcegui4/netk-pager.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/netk-pager.h	2007-04-07 22:28:50.334477679 +0700
@@ -56,9 +56,8 @@
 }
 NetkPagerDisplayMode;
 
-GType
-netk_pager_get_type (void)
-    G_GNUC_CONST;
+G_GNUC_CONST GType
+netk_pager_get_type (void);
 
      GtkWidget *netk_pager_new (NetkScreen * screen);
 
diff -ur libxfcegui4-orig/libxfcegui4/netk-screen.h libxfcegui4/libxfcegui4/netk-screen.h
--- libxfcegui4-orig/libxfcegui4/netk-screen.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/netk-screen.h	2007-04-07 22:34:12.185317476 +0700
@@ -97,9 +97,8 @@
     void (*pad8) (void);
 };
 
-GType
-netk_screen_get_type (void)
-    G_GNUC_CONST;
+G_GNUC_CONST GType
+netk_screen_get_type (void);
 
      NetkScreen *netk_screen_get_default (void);
      NetkScreen *netk_screen_get (int index);
diff -ur libxfcegui4-orig/libxfcegui4/netk-tasklist.h libxfcegui4/libxfcegui4/netk-tasklist.h
--- libxfcegui4-orig/libxfcegui4/netk-tasklist.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/netk-tasklist.h	2007-04-07 22:34:33.889611553 +0700
@@ -56,9 +56,8 @@
 }
 NetkTasklistGroupingType;
 
-GType
-netk_tasklist_get_type (void)
-    G_GNUC_CONST;
+G_GNUC_CONST GType
+netk_tasklist_get_type (void);
 
      GtkWidget *netk_tasklist_new (NetkScreen * screen);
      void netk_tasklist_set_screen (NetkTasklist * tasklist,
diff -ur libxfcegui4-orig/libxfcegui4/netk-window.h libxfcegui4/libxfcegui4/netk-window.h
--- libxfcegui4-orig/libxfcegui4/netk-window.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/netk-window.h	2007-04-07 22:35:25.986798069 +0700
@@ -124,7 +124,7 @@
     void (*geometry_changed) (NetkWindow * window);
 };
 
-GType netk_window_get_type (void) G_GNUC_CONST;
+G_GNUC_CONST GType netk_window_get_type (void);
 
 NetkWindow *netk_window_get (gulong xwindow);
 
diff -ur libxfcegui4-orig/libxfcegui4/netk-workspace.h libxfcegui4/libxfcegui4/netk-workspace.h
--- libxfcegui4-orig/libxfcegui4/netk-workspace.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/netk-workspace.h	2007-04-07 22:35:45.139192906 +0700
@@ -49,9 +49,8 @@
     void (*name_changed) (NetkWorkspace * space);
 };
 
-GType
-netk_workspace_get_type (void)
-    G_GNUC_CONST;
+G_GNUC_CONST GType
+netk_workspace_get_type (void);
 
      int netk_workspace_get_number (NetkWorkspace * space);
 
diff -ur libxfcegui4-orig/libxfcegui4/xfce-appmenuitem.h libxfcegui4/libxfcegui4/xfce-appmenuitem.h
--- libxfcegui4-orig/libxfcegui4/xfce-appmenuitem.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/xfce-appmenuitem.h	2007-04-07 22:41:49.642459817 +0700
@@ -52,9 +52,9 @@
 	GtkImageMenuItemClass parent_class;
 };
 
-GType xfce_app_menu_item_get_type                     () G_GNUC_CONST;
+G_GNUC_CONST GType xfce_app_menu_item_get_type();
 
-GtkWidget *xfce_app_menu_item_new                     ();
+GtkWidget *xfce_app_menu_item_new();
 
 GtkWidget *xfce_app_menu_item_new_with_label          (const gchar *label);
 
diff -ur libxfcegui4-orig/libxfcegui4/xfce-colorbutton.h libxfcegui4/libxfcegui4/xfce-colorbutton.h
--- libxfcegui4-orig/libxfcegui4/xfce-colorbutton.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/xfce-colorbutton.h	2007-04-07 22:42:03.666775706 +0700
@@ -27,7 +27,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #ifndef XFCE_DISABLE_DEPRECATED
 
@@ -60,6 +60,6 @@
 
 #endif  /* ifndef XFCE_DISABLE_DEPRECATED */
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif  /* ifndef __XFCE_COLOR_BUTTON_H__ */
diff -ur libxfcegui4-orig/libxfcegui4/xfce-gdk-extensions.h libxfcegui4/libxfcegui4/xfce-gdk-extensions.h
--- libxfcegui4-orig/libxfcegui4/xfce-gdk-extensions.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/xfce-gdk-extensions.h	2007-04-07 22:43:04.492021320 +0700
@@ -24,14 +24,16 @@
 
 #include <gdk/gdk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
-gchar*      xfce_gdk_display_get_fullname (GdkDisplay *display) G_GNUC_CONST;
+G_GNUC_CONST gchar*
+xfce_gdk_display_get_fullname (GdkDisplay *display);
 GdkScreen*  xfce_gdk_display_locate_monitor_with_pointer (GdkDisplay *display,
                                                           gint       *monitor_return);
 
 
-gchar *xfce_gdk_screen_get_fullname (GdkScreen *screen) G_GNUC_CONST;
+G_GNUC_CONST gchar*
+xfce_gdk_screen_get_fullname (GdkScreen *screen);
 
 
 gboolean xfce_gdk_spawn_on_screen (GdkScreen           *screen,
@@ -61,6 +63,6 @@
 				                                        const gchar  *command_line,
                                         				GError      **error);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_GDK_EXTENSIONS_H__ */
diff -ur libxfcegui4-orig/libxfcegui4/xfce-gtk-extensions.h libxfcegui4/libxfcegui4/xfce-gtk-extensions.h
--- libxfcegui4-orig/libxfcegui4/xfce-gtk-extensions.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/xfce-gtk-extensions.h	2007-04-07 22:43:12.524096919 +0700
@@ -24,7 +24,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 void xfce_gtk_window_center_on_monitor (GtkWindow *window,
                                         GdkScreen *screen,
@@ -32,7 +32,7 @@
 
 void xfce_gtk_window_center_on_monitor_with_pointer (GtkWindow *window);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_GTK_EXTENSIONS_H__ */
 
diff -ur libxfcegui4-orig/libxfcegui4/xfce-heading.c libxfcegui4/libxfcegui4/xfce-heading.c
--- libxfcegui4-orig/libxfcegui4/xfce-heading.c	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/xfce-heading.c	2007-04-07 22:28:50.334809131 +0700
@@ -87,7 +87,7 @@
 
 
 
-G_DEFINE_TYPE (XfceHeading, xfce_heading, GTK_TYPE_WIDGET);
+G_DEFINE_TYPE (XfceHeading, xfce_heading, GTK_TYPE_WIDGET)
 
 
 
diff -ur libxfcegui4-orig/libxfcegui4/xfce-heading.h libxfcegui4/libxfcegui4/xfce-heading.h
--- libxfcegui4-orig/libxfcegui4/xfce-heading.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/xfce-heading.h	2007-04-07 22:44:11.845294860 +0700
@@ -23,7 +23,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceHeadingPrivate XfceHeadingPrivate;
 typedef struct _XfceHeadingClass   XfceHeadingClass;
@@ -57,9 +57,9 @@
   XfceHeadingPrivate *priv;
 };
 
-GType                 xfce_heading_get_type      (void) G_GNUC_CONST;
+G_GNUC_CONST GType xfce_heading_get_type(void);
 
-GtkWidget            *xfce_heading_new           (void) G_GNUC_MALLOC;
+G_GNUC_MALLOC GtkWidget *xfce_heading_new(void);
 
 GdkPixbuf            *xfce_heading_get_icon      (XfceHeading *heading);
 void                  xfce_heading_set_icon      (XfceHeading *heading,
@@ -77,6 +77,6 @@
 void                  xfce_heading_set_title     (XfceHeading *heading,
                                                   const gchar *title);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_HEADING_H__ */
diff -ur libxfcegui4-orig/libxfcegui4/xfce-icontheme.h libxfcegui4/libxfcegui4/xfce-icontheme.h
--- libxfcegui4-orig/libxfcegui4/xfce-icontheme.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/xfce-icontheme.h	2007-04-07 22:50:43.853373857 +0700
@@ -60,48 +60,67 @@
 
 #ifndef XFCE_DISABLE_DEPRECATED
 
-GType xfce_icon_theme_get_type          () G_GNUC_CONST;
+G_GNUC_CONST GType xfce_icon_theme_get_type();
 
-XfceIconTheme *xfce_icon_theme_get_for_screen
-                                        (GdkScreen *screen) G_GNUC_DEPRECATED;
+G_GNUC_DEPRECATED XfceIconTheme*
+xfce_icon_theme_get_for_screen(GdkScreen *screen);
 
-gchar *xfce_icon_theme_lookup           (XfceIconTheme *icon_theme,
-                                         const gchar *icon_name,
-                                         gint icon_size) G_GNUC_DEPRECATED;
-gchar *xfce_icon_theme_lookup_list      (XfceIconTheme *icon_theme,
-                                         GList *icon_names,
-                                         gint icon_size) G_GNUC_DEPRECATED;
-gchar *xfce_icon_theme_lookup_category  (XfceIconTheme *icon_theme,
-                                         XfceIconThemeCategory category,
-                                         gint icon_size) G_GNUC_DEPRECATED;
-
-GdkPixbuf *xfce_icon_theme_load         (XfceIconTheme *icon_theme,
-                                         const gchar *icon_name,
-                                         gint icon_size) G_GNUC_DEPRECATED;
-GdkPixbuf *xfce_icon_theme_load_list    (XfceIconTheme *icon_theme,
-                                         GList *icon_names,
-                                         gint icon_size) G_GNUC_DEPRECATED;
-GdkPixbuf *xfce_icon_theme_load_category(XfceIconTheme *icon_theme,
-                                         XfceIconThemeCategory category,
-                                         gint icon_size) G_GNUC_DEPRECATED;
-
-GList *xfce_icon_theme_get_search_path  (XfceIconTheme *icon_theme) G_GNUC_DEPRECATED;
-void xfce_icon_theme_set_search_path    (XfceIconTheme *icon_theme,
-                                         GList *search_paths) G_GNUC_DEPRECATED;
-void xfce_icon_theme_prepend_search_path(XfceIconTheme *icon_theme,
-                                         const gchar *search_path) G_GNUC_DEPRECATED;
-void xfce_icon_theme_append_search_path (XfceIconTheme *icon_theme,
-                                         const gchar *search_path) G_GNUC_DEPRECATED;
-
-XfceIconThemeCategory xfce_icon_theme_register_category
-                                        (XfceIconTheme *icon_theme,
-                                         GList *icon_names) G_GNUC_DEPRECATED;
-void xfce_icon_theme_unregister_category(XfceIconTheme *icon_theme,
-                                         XfceIconThemeCategory category) G_GNUC_DEPRECATED;
-
-void xfce_icon_theme_set_use_svg        (XfceIconTheme *icon_theme,
-                                         gboolean use_svg) G_GNUC_DEPRECATED;
-gboolean xfce_icon_theme_get_use_svg    (XfceIconTheme *icon_theme) G_GNUC_DEPRECATED;
+G_GNUC_DEPRECATED gchar*
+xfce_icon_theme_lookup(XfceIconTheme *icon_theme,
+                       const gchar *icon_name,
+                       gint icon_size);
+
+G_GNUC_DEPRECATED gchar*
+xfce_icon_theme_lookup_list(XfceIconTheme *icon_theme,
+                            GList *icon_names,
+                            gint icon_size);
+
+G_GNUC_DEPRECATED gchar*
+xfce_icon_theme_lookup_category(XfceIconTheme *icon_theme,
+                                XfceIconThemeCategory category,
+                                gint icon_size);
+
+G_GNUC_DEPRECATED GdkPixbuf*
+xfce_icon_theme_load(XfceIconTheme *icon_theme, const gchar *icon_name,
+                     gint icon_size);
+
+G_GNUC_DEPRECATED GdkPixbuf*
+xfce_icon_theme_load_list(XfceIconTheme *icon_theme,
+                          GList *icon_names,
+                          gint icon_size);
+
+G_GNUC_DEPRECATED GdkPixbuf*
+xfce_icon_theme_load_category(XfceIconTheme *icon_theme,
+                              XfceIconThemeCategory category,
+                              gint icon_size);
+
+G_GNUC_DEPRECATED GList*
+xfce_icon_theme_get_search_path(XfceIconTheme *icon_theme);
+
+G_GNUC_DEPRECATED void
+xfce_icon_theme_set_search_path(XfceIconTheme *icon_theme,GList *search_paths);
+
+G_GNUC_DEPRECATED void
+xfce_icon_theme_prepend_search_path(XfceIconTheme *icon_theme,
+                                    const gchar *search_path);
+
+G_GNUC_DEPRECATED void
+xfce_icon_theme_append_search_path (XfceIconTheme *icon_theme,
+                                    const gchar *search_path);
+
+G_GNUC_DEPRECATED XfceIconThemeCategory
+xfce_icon_theme_register_category(XfceIconTheme *icon_theme,
+                                  GList *icon_names);
+
+G_GNUC_DEPRECATED void
+xfce_icon_theme_unregister_category(XfceIconTheme *icon_theme,
+                                    XfceIconThemeCategory category);
+
+G_GNUC_DEPRECATED void
+xfce_icon_theme_set_use_svg(XfceIconTheme *icon_theme, gboolean use_svg);
+
+G_GNUC_DEPRECATED gboolean
+xfce_icon_theme_get_use_svg(XfceIconTheme *icon_theme);
 
 #endif  /* !defined(XFCE_DISABLE_DEPRECATED) */
 
diff -ur libxfcegui4-orig/libxfcegui4/xfce-titled-dialog.c libxfcegui4/libxfcegui4/xfce-titled-dialog.c
--- libxfcegui4-orig/libxfcegui4/xfce-titled-dialog.c	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/xfce-titled-dialog.c	2007-04-07 22:28:50.335016924 +0700
@@ -70,7 +70,7 @@
 
 
 
-G_DEFINE_TYPE (XfceTitledDialog, xfce_titled_dialog, GTK_TYPE_DIALOG);
+G_DEFINE_TYPE (XfceTitledDialog, xfce_titled_dialog, GTK_TYPE_DIALOG)
 
 
 
diff -ur libxfcegui4-orig/libxfcegui4/xfce-titled-dialog.h libxfcegui4/libxfcegui4/xfce-titled-dialog.h
--- libxfcegui4-orig/libxfcegui4/xfce-titled-dialog.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/xfce-titled-dialog.h	2007-04-07 22:52:52.191729079 +0700
@@ -23,7 +23,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceTitledDialogPrivate XfceTitledDialogPrivate;
 typedef struct _XfceTitledDialogClass   XfceTitledDialogClass;
@@ -57,19 +57,23 @@
   XfceTitledDialogPrivate *priv;
 };
 
-GType                 xfce_titled_dialog_get_type         (void) G_GNUC_CONST;
+G_GNUC_CONST GType xfce_titled_dialog_get_type(void);
 
-GtkWidget            *xfce_titled_dialog_new              (void) G_GNUC_MALLOC;
-GtkWidget            *xfce_titled_dialog_new_with_buttons (const gchar      *title,
-                                                           GtkWindow        *parent,
-                                                           GtkDialogFlags    flags,
-                                                           const gchar      *first_button_text,
-                                                           ...) G_GNUC_MALLOC;
-
-G_CONST_RETURN gchar *xfce_titled_dialog_get_subtitle     (XfceTitledDialog *titled_dialog);
-void                  xfce_titled_dialog_set_subtitle     (XfceTitledDialog *titled_dialog,
-                                                           const gchar      *subtitle);
+G_GNUC_MALLOC GtkWidget* xfce_titled_dialog_new(void);
+G_GNUC_MALLOC GtkWidget*
+xfce_titled_dialog_new_with_buttons (const gchar *title,
+                                     GtkWindow        *parent,
+                                     GtkDialogFlags    flags,
+                                     const gchar      *first_button_text,
+                                     ...);
+
+G_CONST_RETURN gchar*
+xfce_titled_dialog_get_subtitle     (XfceTitledDialog *titled_dialog);
+
+void
+xfce_titled_dialog_set_subtitle     (XfceTitledDialog *titled_dialog,
+                                    const gchar      *subtitle);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_TITLED_DIALOG_H__ */
diff -ur libxfcegui4-orig/libxfcegui4/xfce_aboutdialog.c libxfcegui4/libxfcegui4/xfce_aboutdialog.c
--- libxfcegui4-orig/libxfcegui4/xfce_aboutdialog.c	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/xfce_aboutdialog.c	2007-04-07 22:28:50.335476842 +0700
@@ -543,7 +543,7 @@
 
 
 
-G_DEFINE_TYPE (XfceAboutDialog, xfce_about_dialog, GTK_TYPE_DIALOG);
+G_DEFINE_TYPE (XfceAboutDialog, xfce_about_dialog, GTK_TYPE_DIALOG)
 
 
 
diff -ur libxfcegui4-orig/libxfcegui4/xfce_aboutdialog.h libxfcegui4/libxfcegui4/xfce_aboutdialog.h
--- libxfcegui4-orig/libxfcegui4/xfce_aboutdialog.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/xfce_aboutdialog.h	2007-04-07 22:39:26.903452238 +0700
@@ -26,7 +26,7 @@
 
 #include <libxfce4util/libxfce4util.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #define XFCE_COPYRIGHT_TEXT(years, owner) ("Copyright (c) " years "\n" owner "\nAll rights reserved.")
 
@@ -35,15 +35,18 @@
 
 #define XFCE_TYPE_ABOUT_INFO (xfce_about_info_get_type ())
 
-GType             xfce_about_info_get_type        (void) G_GNUC_CONST;
+G_GNUC_CONST GType
+xfce_about_info_get_type        (void);
 
-XfceAboutInfo    *xfce_about_info_new             (const gchar          *program,
-                                                   const gchar          *version,
-                                                   const gchar          *description,
-                                                   const gchar          *copyright,
-                                                   const gchar          *license) G_GNUC_MALLOC;
+G_GNUC_MALLOC XfceAboutInfo*
+xfce_about_info_new(const gchar          *program,
+                    const gchar          *version,
+                    const gchar          *description,
+                    const gchar          *copyright,
+                    const gchar          *license);
 
-XfceAboutInfo    *xfce_about_info_copy            (const XfceAboutInfo  *info) G_GNUC_MALLOC;
+G_GNUC_MALLOC XfceAboutInfo*
+xfce_about_info_copy(const XfceAboutInfo  *info);
 
 void              xfce_about_info_free            (XfceAboutInfo        *info);
 
@@ -99,18 +102,21 @@
   XfceAboutDialogPrivate *priv;
 };
 
-GType       xfce_about_dialog_get_type (void) G_GNUC_CONST;
+G_GNUC_CONST GType       xfce_about_dialog_get_type (void);
 
-GtkWidget*  xfce_about_dialog_new_with_values (GtkWindow            *parent,
+G_GNUC_MALLOC GtkWidget*
+xfce_about_dialog_new_with_values (GtkWindow            *parent,
                                    const XfceAboutInfo  *info,
-                                   GdkPixbuf            *icon) G_GNUC_MALLOC;
+                                   GdkPixbuf            *icon);
 
 /* old: xfce_about_dialog_new */
-GtkWidget*  xfce_about_dialog_new (GtkWindow            *parent,
-                                   const XfceAboutInfo  *info,
-                                   GdkPixbuf            *icon) G_GNUC_DEPRECATED G_GNUC_MALLOC;
 
-GtkWidget*  xfce_about_dialog_new_empty (void) G_GNUC_MALLOC;
+G_GNUC_DEPRECATED G_GNUC_MALLOC GtkWidget*
+xfce_about_dialog_new (GtkWindow            *parent,
+                       const XfceAboutInfo  *info,
+                       GdkPixbuf            *icon);
+
+G_GNUC_MALLOC GtkWidget*  xfce_about_dialog_new_empty (void);
 void xfce_about_dialog_set_program(XfceAboutDialog* dialog, const gchar *value);
 void xfce_about_dialog_set_version(XfceAboutDialog* dialog, const gchar *value);
 void xfce_about_dialog_set_description(XfceAboutDialog* dialog, const gchar *value);
@@ -131,6 +137,6 @@
 const gchar *xfce_about_dialog_get_homepage(XfceAboutDialog* dialog);
 GdkPixbuf   *xfce_about_dialog_get_icon(XfceAboutDialog* dialog);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif  /* !__LIBXFCEGUI4_XFCE_ABOUTDIALOG_H__ */
diff -ur libxfcegui4-orig/libxfcegui4/xfce_framebox.h libxfcegui4/libxfcegui4/xfce_framebox.h
--- libxfcegui4-orig/libxfcegui4/xfce_framebox.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/xfce_framebox.h	2007-04-07 22:40:18.144569494 +0700
@@ -47,7 +47,7 @@
 
 #ifndef XFCE_DISABLE_DEPRECATED
 
-    GType xfce_framebox_get_type        (void) G_GNUC_CONST;
+    G_GNUC_CONST GType xfce_framebox_get_type(void);
     GtkWidget *xfce_framebox_new        (const gchar *text,
                                          gboolean indent);
     
diff -ur libxfcegui4-orig/libxfcegui4/xinerama.h libxfcegui4/libxfcegui4/xinerama.h
--- libxfcegui4-orig/libxfcegui4/xinerama.h	2007-04-07 09:34:33.000000000 +0700
+++ libxfcegui4/libxfcegui4/xinerama.h	2007-04-07 22:53:10.152152165 +0700
@@ -24,7 +24,7 @@
 #ifndef __XINERAMA_H__
 #define __XINERAMA_H__
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #ifndef XFCE_DISABLE_DEPRECATED
 
@@ -46,6 +46,6 @@
 
 #endif /* XFCE_DISABLE_DEPRECATED */
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* __XINERAMA_H__ */
