--- libxfce4mcs-4.4.0-orig/libxfce4mcs/mcs-manager.h	Sat Jan 20 15:54:53 2007
+++ libxfce4mcs-4.4.0/libxfce4mcs/mcs-manager.h	Sun Mar  4 16:58:45 2007
@@ -49,7 +49,7 @@
 #include <libxfce4mcs/mcs-common.h>
 
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _McsManager McsManager;
 
@@ -85,6 +85,6 @@
 McsChannel *mcs_manager_add_channel_from_file(McsManager *manager, const gchar *channel_name, const gchar *filename);
 gboolean mcs_manager_save_channel_to_file(McsManager *manager, const gchar *channel_name, const gchar *filename);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* MCS_MANAGER_H */
