Index: 2.2.0/mediastreamer.pc.in
===================================================================
--- 2.2.0/mediastreamer.pc.in	(revision 2601)
+++ 2.2.0/mediastreamer.pc.in	(working copy)
@@ -5,7 +5,6 @@
 
 Name: mediastreamer 
 Description: A mediastreaming library for telephony applications
-Depends: ortp
 Version: @MEDIASTREAMER_VERSION@
 Libs: -L@libdir@ -lmediastreamer
 Cflags: -I@includedir@ 
