--- mediastreamer-2.1.1.orig/src/nowebcam.h	2007-03-09 15:56:35.000000000 -0500
+++ mediastreamer-2.1.1/src/nowebcam.h	2008-05-16 21:07:46.000000000 -0400
@@ -21,8 +21,8 @@
 #ifndef nowebcam_h
 #define nowebcam_h
 
-mblk_t *ms_load_jpeg_as_yuv(const char *path, MSVideoSize *reqsize);
-mblk_t *ms_load_nowebcam(MSVideoSize *reqsize, int idx);
+ortp_mblk_t *ms_load_jpeg_as_yuv(const char *path, MSVideoSize *reqsize);
+ortp_mblk_t *ms_load_nowebcam(MSVideoSize *reqsize, int idx);
 
 #endif
 
