--- mplayer-checkout-2009-04-26/configure.orig	2009-04-26 09:45:09.000000000 +0530
+++ mplayer-checkout-2009-04-26/configure	2009-04-26 22:47:02.183992964 +0530
@@ -7922,6 +7922,7 @@
 if test "$_real" = yes || test "$_xanim" = yes && ! win32 && ! qnx && ! darwin && ! os2 ; then
   _ld_dl_dynamic='-rdynamic'
 fi
+sunos && _ld_dl_dynamic='-dynamic'
 
 extra_ldflags="$extra_ldflags $_ld_pthread $_ld_dl $_ld_dl_dynamic"
 bsdos && extra_ldflags="$extra_ldflags -ldvd"
