--- ortp-0.16.0/src/tests/rtpsend_stupid.c.orig	2009-05-14 11:48:51.834068552 +0530
+++ ortp-0.16.0/src/tests/rtpsend_stupid.c	2009-05-14 11:49:17.519770209 +0530
@@ -27,7 +27,7 @@
 #include <stdio.h>
 #endif
 /*defined in library, but not declared in public headers (this method is only useful for tests)*/
-extern int __rtp_session_sendm_with_ts(RtpSession *session, mblk_t *packet, uint32_t packet_ts, uint32_t send_ts);
+extern int __rtp_session_sendm_with_ts(RtpSession *session, ortp_mblk_t *packet, uint32_t packet_ts, uint32_t send_ts);
 
 int runcond=1;
 
@@ -105,7 +105,7 @@
 	signal(SIGINT,stophandler);
 	while( ((i=fread(buffer,1,160,infile))>0) && (runcond) )
 	{
-		mblk_t *m=rtp_session_create_packet(session,RTP_FIXED_HEADER_SIZE,buffer,i);
+		ortp_mblk_t *m=rtp_session_create_packet(session,RTP_FIXED_HEADER_SIZE,buffer,i);
 		__rtp_session_sendm_with_ts(session,m,packet_ts,send_ts);
 		packet_ts+=160;
 		if ((send_ts+send_ts_inc)<=packet_ts){
