--- subversion-1.5.0/build/ac-macros/neon.m4.orig	2009-01-09 18:35:54.265620561 +0530
+++ subversion-1.5.0/build/ac-macros/neon.m4	2009-01-09 18:36:27.502049737 +0530
@@ -153,7 +153,7 @@
            test "$svn_allowed_neon" = "any"; then
             svn_allowed_neon_on_system="yes"
             SVN_NEON_INCLUDES=[`$neon_config --cflags | sed -e 's/-D[^ ]*//g'`]
-            NEON_LIBS=`$neon_config --la-file`
+            NEON_LIBS=`$neon_config --libs`
             CFLAGS=["$CFLAGS `$neon_config --cflags | sed -e 's/-I[^ ]*//g'`"]
             svn_lib_neon="yes"
             break
