--- vegastrike-0.4.3/launcher/saveinterface.cpp~	2006-02-10 10:54:30.000000000 +0100
+++ vegastrike-0.4.3/launcher/saveinterface.cpp	2006-02-10 11:11:33.000000000 +0100
@@ -213,7 +213,7 @@
 #ifdef _WIN32
   GoToParentDir();
 #endif
-  int player = my_mission.rfind ("player");
+  std::string::size_type player = my_mission.rfind ("player");
   if (player>0&&player!=std::string::npos) {
    char  num [4]={'-','m',(*(my_mission.begin()+(player-1))),'\0'};
    printf ("vegastrike %s %s",num,my_mission.c_str());
