--- vegastrike-source-0.5.0/src/networking/lowlevel/vsnet_socketset.cpp.orig	2009-11-15 18:08:39.754877778 +0530
+++ vegastrike-source-0.5.0/src/networking/lowlevel/vsnet_socketset.cpp	2009-11-15 18:09:37.417401706 +0530
@@ -180,7 +180,7 @@
     
     if( !_client_mgr.expired() )
     {
-        boost::shared_ptr<VsnetDownload::Client::Manager> mgr( boost::make_shared(_client_mgr) );
+        boost::shared_ptr<VsnetDownload::Client::Manager> mgr( _client_mgr.lock() );
         if( (bool)mgr )
         {
             mgr->lower_check_queues( );
@@ -189,7 +189,7 @@
 
     if( !_server_mgr.expired() )
     {
-        boost::shared_ptr<VsnetDownload::Server::Manager> mgr( boost::make_shared(_server_mgr) );
+        boost::shared_ptr<VsnetDownload::Server::Manager> mgr( _server_mgr.lock() );
         if( (bool)mgr )
         {
             mgr->lower_check_queues( );
