--- vlc-0.9.9a/modules/audio_output/oss.c.orig	2009-04-04 20:19:11.000000000 +0530
+++ vlc-0.9.9a/modules/audio_output/oss.c	2009-05-10 18:54:28.271097315 +0530
@@ -33,6 +33,9 @@
 
 #include <errno.h>                                                 /* ENOMEM */
 #include <fcntl.h>                                       /* open(), O_WRONLY */
+#if defined(sun) || defined(__sun)
+#include <sys/file.h>						  /* FNDELAY */
+#endif
 #include <sys/ioctl.h>                                            /* ioctl() */
 #include <unistd.h>                                      /* write(), close() */
 
