--- vlc-0.9.9a/modules/video_chroma/i420_rgb16.c.orig	2009-05-10 20:11:51.778578777 +0530
+++ vlc-0.9.9a/modules/video_chroma/i420_rgb16.c	2009-05-10 20:33:00.824551608 +0530
@@ -35,9 +35,10 @@
 #include <vlc_vout.h>
 
 #include "i420_rgb.h"
-#if defined (MODULE_NAME_IS_i420_rgb)
+#if defined (MODULE_NAME_IS_i420_rgb) || defined(__PIC__)
 #   include "i420_rgb_c.h"
-#elif defined (MODULE_NAME_IS_i420_rgb_mmx)
+#endif
+#if defined (MODULE_NAME_IS_i420_rgb_mmx)
 #   include "i420_rgb_mmx.h"
 #elif defined (MODULE_NAME_IS_i420_rgb_sse2)
 #   include "i420_rgb_mmx.h"
@@ -46,7 +47,7 @@
 static void SetOffset( int, int, int, int, bool *,
                        unsigned int *, int * );
 
-#if defined (MODULE_NAME_IS_i420_rgb)
+#if defined (MODULE_NAME_IS_i420_rgb) || defined(__PIC__)
 /*****************************************************************************
  * I420_RGB16: color YUV 4:2:0 to RGB 16 bpp with dithering
  *****************************************************************************
@@ -209,7 +210,7 @@
  *  - output: 1 line
  *****************************************************************************/
 
-#if defined (MODULE_NAME_IS_i420_rgb)
+#if defined (MODULE_NAME_IS_i420_rgb) || defined(__PIC__)
 
 void I420_RGB16( filter_t *p_filter, picture_t *p_src,
                                          picture_t *p_dest )
@@ -815,7 +816,7 @@
  *  - output: 1 line
  *****************************************************************************/
 
-#if defined (MODULE_NAME_IS_i420_rgb)
+#if defined (MODULE_NAME_IS_i420_rgb) || defined(__PIC__)
 
 void I420_RGB32( filter_t *p_filter, picture_t *p_src,
                                          picture_t *p_dest )
--- vlc-0.9.9a/modules/video_chroma/i420_rgb.h.orig	2009-05-10 20:26:31.297477909 +0530
+++ vlc-0.9.9a/modules/video_chroma/i420_rgb.h	2009-05-10 20:31:11.669311516 +0530
@@ -35,7 +35,7 @@
     uint8_t  *p_buffer;
     int *p_offset;
 
-#ifdef MODULE_NAME_IS_i420_rgb
+#if defined(MODULE_NAME_IS_i420_rgb) || defined(__PIC__)
     /**< Pre-calculated conversion tables */
     void *p_base;                      /**< base for all conversion tables */
     uint8_t   *p_rgb8;                 /**< RGB 8 bits table */
@@ -55,7 +55,7 @@
 /*****************************************************************************
  * Prototypes
  *****************************************************************************/
-#ifdef MODULE_NAME_IS_i420_rgb
+#if defined(MODULE_NAME_IS_i420_rgb)
 void I420_RGB8         ( filter_t *, picture_t *, picture_t * );
 void I420_RGB16_dither ( filter_t *, picture_t *, picture_t * );
 void I420_RGB16        ( filter_t *, picture_t *, picture_t * );
