--- gpac/Makefile.orig	2008-02-21 06:11:06.459784059 -0800
+++ gpac/Makefile	2008-02-21 06:11:50.170896763 -0800
@@ -69,7 +69,7 @@
 	ln -sf libgpac-$(VERSION).$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/lib/libgpac.$(DYN_LIB_SUFFIX)
 else
 	install $(INSTFLAGS) -m 755 bin/gcc/libgpac.$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/lib/libgpac-$(VERSION).$(DYN_LIB_SUFFIX)
-	ln -sf libgpac-$(VERSION).$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/lib/libgpac.$(DYN_LIB_SUFFIX)
+	(cd $(DESTDIR)$(prefix)/lib;/usr/bin/ln -sf libgpac-$(VERSION).$(DYN_LIB_SUFFIX) libgpac.$(DYN_LIB_SUFFIX))
 	ldconfig || true
 endif
 endif
