--- jack-audio-connection-kit-0.109.2/drivers/dummy/dummy_driver.c.orig	2008-01-30 23:53:56.000000000 +0530
+++ jack-audio-connection-kit-0.109.2/drivers/dummy/dummy_driver.c	2009-04-26 20:46:32.274481843 +0530
@@ -44,8 +44,10 @@
 FakeVideoSync( dummy_driver_t *driver )
 {
         #define VIDEO_SYNC_PERIOD (48000 / 30)
-        static int vidCounter = VIDEO_SYNC_PERIOD;
-        
+        static int vidCounter;
+
+        vidCounter = VIDEO_SYNC_PERIOD;
+
         int period = driver->period_size;
         jack_position_t *position = &driver->engine->control->current_time;
 
