--- usr/share/config/kdm/Xsession.orig	2008-02-22 20:36:13.680632000 +0530
+++ usr/share/config/kdm/Xsession	2008-02-22 20:38:20.952647000 +0530
@@ -56,10 +56,10 @@
     exec $HOME/.xsession
     ;;
   default)
-    exec /usr/bin/startkde
+    exec /usr/bin/ctrun -i none -l child /usr/bin/startkde
     ;;
   *)
-    eval exec "$session"
+    eval exec /usr/bin/ctrun -i none -l child "$session"
     ;;
 esac
 exec xmessage -center -buttons OK:0 -default OK "Sorry, cannot execute $session. Check $DESKTOP_SESSION.desktop."
