--- exo-0.3.4/exo-support/xfce-titled-dialog.h.fixgccism	2007-12-09 21:06:33.007411853 +0100
+++ exo-0.3.4/exo-support/xfce-titled-dialog.h	2007-12-09 21:07:50.420279798 +0100
@@ -23,7 +23,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceTitledDialogPrivate XfceTitledDialogPrivate;
 typedef struct _XfceTitledDialogClass   XfceTitledDialogClass;
@@ -57,19 +57,19 @@
   XfceTitledDialogPrivate *priv;
 };
 
-GType                 xfce_titled_dialog_get_type         (void) G_GNUC_CONST;
+G_GNUC_CONST GType                 xfce_titled_dialog_get_type         (void);
 
-GtkWidget            *xfce_titled_dialog_new              (void) G_GNUC_MALLOC;
-GtkWidget            *xfce_titled_dialog_new_with_buttons (const gchar      *title,
+G_GNUC_MALLOC GtkWidget            *xfce_titled_dialog_new              (void);
+G_GNUC_MALLOC GtkWidget            *xfce_titled_dialog_new_with_buttons (const gchar      *title,
                                                            GtkWindow        *parent,
                                                            GtkDialogFlags    flags,
                                                            const gchar      *first_button_text,
-                                                           ...) G_GNUC_MALLOC;
+                                                           ...);
 
 G_CONST_RETURN gchar *xfce_titled_dialog_get_subtitle     (XfceTitledDialog *titled_dialog);
 void                  xfce_titled_dialog_set_subtitle     (XfceTitledDialog *titled_dialog,
                                                            const gchar      *subtitle);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_TITLED_DIALOG_H__ */
--- exo-0.3.4/exo-support/xfce-heading.c.fixgccism	2007-12-09 21:04:26.297259873 +0100
+++ exo-0.3.4/exo-support/xfce-heading.c	2007-12-09 21:05:35.205260905 +0100
@@ -87,7 +87,7 @@
 
 
 
-G_DEFINE_TYPE (XfceHeading, xfce_heading, GTK_TYPE_WIDGET);
+G_DEFINE_TYPE (XfceHeading, xfce_heading, GTK_TYPE_WIDGET)
 
 
 
--- exo-0.3.4/exo-support/xfce-titled-dialog.c.fixgccism	2007-12-09 21:06:00.567619884 +0100
+++ exo-0.3.4/exo-support/xfce-titled-dialog.c	2007-12-09 21:06:24.461895082 +0100
@@ -70,7 +70,7 @@
 
 
 
-G_DEFINE_TYPE (XfceTitledDialog, xfce_titled_dialog, GTK_TYPE_DIALOG);
+G_DEFINE_TYPE (XfceTitledDialog, xfce_titled_dialog, GTK_TYPE_DIALOG)
 
 
 
--- exo-0.3.4/exo-support/xfce-heading.h.fixgccism	2007-12-09 21:03:08.065817383 +0100
+++ exo-0.3.4/exo-support/xfce-heading.h	2007-12-09 21:04:12.934477195 +0100
@@ -23,7 +23,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _XfceHeadingPrivate XfceHeadingPrivate;
 typedef struct _XfceHeadingClass   XfceHeadingClass;
@@ -57,9 +57,9 @@
   XfceHeadingPrivate *priv;
 };
 
-GType                 xfce_heading_get_type      (void) G_GNUC_CONST;
+G_GNUC_CONST GType                 xfce_heading_get_type      (void);
 
-GtkWidget            *xfce_heading_new           (void) G_GNUC_MALLOC;
+G_GNUC_MALLOC GtkWidget            *xfce_heading_new           (void);
 
 GdkPixbuf            *xfce_heading_get_icon      (XfceHeading *heading);
 void                  xfce_heading_set_icon      (XfceHeading *heading,
@@ -77,6 +77,6 @@
 void                  xfce_heading_set_title     (XfceHeading *heading,
                                                   const gchar *title);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__XFCE_HEADING_H__ */
--- exo-0.3.4/exo-mount/exo-mount-utils.h.fixgccism	2007-12-09 21:00:26.059511086 +0100
+++ exo-0.3.4/exo-mount/exo-mount-utils.h	2007-12-09 21:01:00.404820325 +0100
@@ -22,11 +22,11 @@
 
 #include <exo/exo.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
-gboolean exo_mount_utils_is_mounted (const gchar *device_file,
-                                     gboolean    *readonly_return) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL gboolean exo_mount_utils_is_mounted (const gchar *device_file,
+                                     gboolean    *readonly_return);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_MOUNT_UTILS_H__ */
--- exo-0.3.4/exo-mount/exo-mount-hal.h.fixgccism	2007-12-09 20:55:18.456843783 +0100
+++ exo-0.3.4/exo-mount/exo-mount-hal.h	2007-12-09 20:59:50.083042297 +0100
@@ -22,30 +22,39 @@
 
 #include <exo/exo.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoMountHalDevice ExoMountHalDevice;
 
+G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT 
 ExoMountHalDevice *exo_mount_hal_device_from_udi    (const gchar       *udi,
-                                                     GError           **error) G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+                                                     GError           **error);
+
+G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT
 ExoMountHalDevice *exo_mount_hal_device_from_file   (const gchar       *file,
-                                                     GError           **error) G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+                                                     GError           **error);
 
-void               exo_mount_hal_device_free        (ExoMountHalDevice *device) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL void               exo_mount_hal_device_free        (ExoMountHalDevice *device);
 
-gchar             *exo_mount_hal_device_get_file    (ExoMountHalDevice *device) G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
-gchar             *exo_mount_hal_device_get_name    (ExoMountHalDevice *device) G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
-gchar             *exo_mount_hal_device_get_icon    (ExoMountHalDevice *device) G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT
+gchar             *exo_mount_hal_device_get_file    (ExoMountHalDevice *device);
+G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT
+gchar             *exo_mount_hal_device_get_name    (ExoMountHalDevice *device);
+G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT
+gchar             *exo_mount_hal_device_get_icon    (ExoMountHalDevice *device);
 
-gboolean           exo_mount_hal_device_is_readonly (ExoMountHalDevice *device) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL gboolean           exo_mount_hal_device_is_readonly (ExoMountHalDevice *device);
 
+G_GNUC_INTERNAL 
 gboolean           exo_mount_hal_device_eject       (ExoMountHalDevice *device,
-                                                     GError           **error) G_GNUC_INTERNAL;
+                                                     GError           **error);
+G_GNUC_INTERNAL
 gboolean           exo_mount_hal_device_mount       (ExoMountHalDevice *device,
-                                                     GError           **error) G_GNUC_INTERNAL;
+                                                     GError           **error);
+G_GNUC_INTERNAL
 gboolean           exo_mount_hal_device_unmount     (ExoMountHalDevice *device,
-                                                     GError           **error) G_GNUC_INTERNAL;
+                                                     GError           **error);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_MOUNT_HAL_H__ */
--- exo-0.3.4/exo-mount/exo-mount-fstab.h.fixgccism	2007-12-09 20:53:14.003640023 +0100
+++ exo-0.3.4/exo-mount/exo-mount-fstab.h	2007-12-09 20:54:42.645935540 +0100
@@ -22,19 +22,19 @@
 
 #include <exo/exo.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
-gboolean exo_mount_fstab_contains (const gchar *device_file) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL gboolean exo_mount_fstab_contains (const gchar *device_file);
 
-gboolean exo_mount_fstab_eject    (const gchar *device_file,
-                                   GError     **error) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL gboolean exo_mount_fstab_eject    (const gchar *device_file,
+                                   GError     **error);
 
-gboolean exo_mount_fstab_mount    (const gchar *device_file,
-                                   GError     **error) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL gboolean exo_mount_fstab_mount    (const gchar *device_file,
+                                   GError     **error);
 
-gboolean exo_mount_fstab_unmount  (const gchar *device_file,
-                                   GError     **error) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL gboolean exo_mount_fstab_unmount  (const gchar *device_file,
+                                   GError     **error);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_MOUNT_FSTAB_H__ */
--- exo-0.3.4/exo-hal/exo-hal.h.fixgccism	2007-12-09 20:33:18.674491913 +0100
+++ exo-0.3.4/exo-hal/exo-hal.h	2007-12-09 22:44:57.156549786 +0100
@@ -22,7 +22,7 @@
 
 #include <glib.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 /* forward declarations for libhal-storage */
 #ifndef LIBHAL_STORAGE_H
@@ -46,18 +46,22 @@
                                                gssize                  len,
                                                gchar                 **end);
 
+G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT
 gchar   *exo_hal_drive_compute_display_name   (struct LibHalContext_s *context,
-                                               struct LibHalDrive_s   *drive) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+                                               struct LibHalDrive_s   *drive);
+G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT 
 GList   *exo_hal_drive_compute_icon_list      (struct LibHalContext_s *context,
-                                               struct LibHalDrive_s   *drive) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+                                               struct LibHalDrive_s   *drive);
 
+G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT 
 gchar   *exo_hal_volume_compute_display_name  (struct LibHalContext_s *context,
                                                struct LibHalVolume_s  *volume,
-                                               struct LibHalDrive_s   *drive) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+                                               struct LibHalDrive_s   *drive);
+G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT 
 GList   *exo_hal_volume_compute_icon_list     (struct LibHalContext_s *context,
                                                struct LibHalVolume_s  *volume,
-                                               struct LibHalDrive_s   *drive) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+                                               struct LibHalDrive_s   *drive);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_HAL_H__ */
--- exo-0.3.4/exo/exo-utils.h.fixgccism	2007-12-09 20:02:42.405460896 +0100
+++ exo-0.3.4/exo/exo-utils.h	2007-12-09 20:04:44.155523095 +0100
@@ -27,14 +27,14 @@
 
 #include <glib.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
-void                    exo_noop        (void) G_GNUC_PURE;
-gint                    exo_noop_one    (void) G_GNUC_PURE;
-gint                    exo_noop_zero   (void) G_GNUC_PURE;
-gpointer                exo_noop_null   (void) G_GNUC_PURE;
-gboolean                exo_noop_true   (void) G_GNUC_PURE;
-gboolean                exo_noop_false  (void) G_GNUC_PURE;
+G_GNUC_PURE void                    exo_noop        (void);
+G_GNUC_PURE gint                    exo_noop_one    (void);
+G_GNUC_PURE gint                    exo_noop_zero   (void);
+G_GNUC_PURE gpointer                exo_noop_null   (void);
+G_GNUC_PURE gboolean                exo_noop_true   (void);
+G_GNUC_PURE gboolean                exo_noop_false  (void);
 G_INLINE_FUNC void      exo_atomic_inc  (gint *value);
 G_INLINE_FUNC gboolean  exo_atomic_dec  (gint *value);
 
@@ -81,6 +81,6 @@
 
 #endif /* (G_CAN_INLINE && __GNUC__ && __i386__ && __OPTIMIZE__) || __EXO_UTILS_C__ */
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_UTILS_H__ */
--- exo-0.3.4/exo/exo-binding.h.fixgccism	2007-12-09 18:51:10.500463255 +0100
+++ exo-0.3.4/exo/exo-binding.h	2007-12-09 18:51:53.258012644 +0100
@@ -30,7 +30,7 @@
 
 #include <glib-object.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoBindingBase    ExoBindingBase;
 typedef struct _ExoBindingLink    ExoBindingLink;
@@ -108,6 +108,6 @@
                                                          const gchar        *property2);
 void               exo_mutual_binding_unbind            (ExoMutualBinding   *binding);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_BINDING_H__ */
--- exo-0.3.4/exo/exo-gtk-extensions.h.fixgccism	2007-12-09 19:11:27.065223553 +0100
+++ exo-0.3.4/exo/exo-gtk-extensions.h	2007-12-09 19:11:43.131792532 +0100
@@ -29,7 +29,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 void      exo_gtk_object_destroy_later                (GtkObject      *object);
 
@@ -40,6 +40,6 @@
 
 void      exo_gtk_file_chooser_add_thumbnail_preview  (GtkFileChooser *chooser);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_GTK_EXTENSIONS_H__ */
--- exo-0.3.4/exo/exo-thumbnail.c.fixgccism	2007-12-09 19:39:13.875022697 +0100
+++ exo-0.3.4/exo/exo-thumbnail.c	2007-12-09 19:40:54.850668238 +0100
@@ -70,7 +70,7 @@
 static GdkPixbuf *exo_thumbnail_load (const gchar *thumbnail_path,
                                       const gchar *uri,
                                       time_t       mtime,
-                                      GError     **error) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;;
+                                      GError     **error) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
 static gboolean   exo_thumbnail_save (GdkPixbuf   *thumbnail,
                                       const gchar *thumbnail_path,
                                       const gchar *uri,
--- exo-0.3.4/exo/exo-icon-bar.h.fixgccism	2007-12-09 19:12:41.525517654 +0100
+++ exo-0.3.4/exo/exo-icon-bar.h	2007-12-09 19:12:59.323230309 +0100
@@ -29,7 +29,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #define EXO_TYPE_ICON_BAR             (exo_icon_bar_get_type ())
 #define EXO_ICON_BAR(obj)             (G_TYPE_CHECK_INSTANCE_CAST ((obj), EXO_TYPE_ICON_BAR, ExoIconBar))
@@ -96,7 +96,7 @@
 void            exo_icon_bar_set_active_iter    (ExoIconBar     *icon_bar,
                                                  GtkTreeIter    *iter);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_ICON_BAR_H__ */
 
--- exo-0.3.4/exo/exo-pango-extensions.h.fixgccism	2007-12-09 19:35:38.113956524 +0100
+++ exo-0.3.4/exo/exo-pango-extensions.h	2007-12-09 19:35:54.296893473 +0100
@@ -33,7 +33,7 @@
 
 #include <pango/pango.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 /* define ExoPangoEllipsizeMode for backward compatibility */
 typedef PangoEllipsizeMode ExoPangoEllipsizeMode;
@@ -51,7 +51,7 @@
                                                gint               width,
                                                PangoEllipsizeMode mode);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_PANGO_EXTENSIONS_H__ */
 
--- exo-0.3.4/exo/exo-gobject-extensions.h.fixgccism	2007-12-09 19:10:42.355124067 +0100
+++ exo-0.3.4/exo/exo-gobject-extensions.h	2007-12-09 19:10:59.514216179 +0100
@@ -29,11 +29,11 @@
 
 #include <glib-object.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 gboolean  exo_g_value_transform_negate  (const GValue *src_value,
                                          GValue       *dst_value);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_GOBJECT_EXTENSIONS_H__ */
--- exo-0.3.4/exo/exo-xsession-client.h.fixgccism	2007-12-09 20:09:21.505564821 +0100
+++ exo-0.3.4/exo/exo-xsession-client.h	2007-12-09 20:10:23.736031299 +0100
@@ -29,7 +29,7 @@
 
 #include <gdk/gdk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoXsessionClientPrivate ExoXsessionClientPrivate;
 typedef struct _ExoXsessionClientClass   ExoXsessionClientClass;
@@ -61,7 +61,7 @@
   ExoXsessionClientPrivate *priv;
 };
 
-GType              exo_xsession_client_get_type             (void) G_GNUC_CONST;
+G_GNUC_CONST GType              exo_xsession_client_get_type             (void);
 
 ExoXsessionClient *exo_xsession_client_new_with_group       (GdkWindow          *leader);
 
@@ -76,6 +76,6 @@
                                                              gchar             **argv,
                                                              gint                argc);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_XSESSION_CLIENT_H__ */
--- exo-0.3.4/exo/exo-toolbars-editor.h.fixgccism	2007-12-09 19:47:09.089449249 +0100
+++ exo-0.3.4/exo/exo-toolbars-editor.h	2007-12-09 19:47:58.318407773 +0100
@@ -30,7 +30,7 @@
 
 #include <exo/exo-toolbars-view.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #define EXO_TYPE_TOOLBARS_EDITOR            (exo_toolbars_editor_get_type ())
 #define EXO_TOOLBARS_EDITOR(obj)            (G_TYPE_CHECK_INSTANCE_CAST ((obj), EXO_TYPE_TOOLBARS_EDITOR, ExoToolbarsEditor))
@@ -62,7 +62,7 @@
 };
 
 
-GType             exo_toolbars_editor_get_type        (void) G_GNUC_CONST;
+G_GNUC_CONST GType             exo_toolbars_editor_get_type        (void);
 GtkWidget        *exo_toolbars_editor_new             (GtkUIManager       *ui_manager);
 GtkWidget        *exo_toolbars_editor_new_with_model  (GtkUIManager       *ui_manager,
                                                        ExoToolbarsModel   *model);
@@ -75,6 +75,6 @@
 void              exo_toolbars_editor_set_ui_manager  (ExoToolbarsEditor  *editor,
                                                        GtkUIManager       *ui_manager);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_TOOLBARS_EDITOR_H__ */
--- exo-0.3.4/exo/exo-ellipsized-label.h.fixgccism	2007-12-09 18:59:01.448930682 +0100
+++ exo-0.3.4/exo/exo-ellipsized-label.h	2007-12-09 18:59:44.025677055 +0100
@@ -32,7 +32,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #define EXO_TYPE_ELLIPSIZED_LABEL            (exo_ellipsized_label_get_type ())
 #define EXO_ELLIPSIZED_LABEL(obj)            (G_TYPE_CHECK_INSTANCE_CAST ((obj), EXO_TYPE_ELLIPSIZED_LABEL, ExoEllipsizedLabel))
@@ -66,7 +66,7 @@
 void               exo_ellipsized_label_set_ellipsize (ExoEllipsizedLabel *label,
                                                        PangoEllipsizeMode  ellipsize);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_ELLIPSIZED_LABEL_H__ */
 
--- exo-0.3.4/exo/exo-md5.h.fixgccism	2007-12-09 19:19:02.507597007 +0100
+++ exo-0.3.4/exo/exo-md5.h	2007-12-09 19:19:28.643254853 +0100
@@ -30,7 +30,7 @@
 
 #include <exo/exo-config.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #define EXO_TYPE_MD5_DIGEST (exo_md5_digest_get_type ())
 
@@ -55,6 +55,6 @@
 gboolean      exo_md5_digest_equal    (gconstpointer       digest1,
                                        gconstpointer       digest2);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_MD5_H__ */
--- exo-0.3.4/exo/exo-thumbnail.h.fixgccism	2007-12-09 19:41:05.188999307 +0100
+++ exo-0.3.4/exo/exo-thumbnail.h	2007-12-09 21:18:43.864870414 +0100
@@ -29,7 +29,7 @@
 
 #include <gdk-pixbuf/gdk-pixbuf.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 /**
  * ExoThumbnailSize:
@@ -44,13 +44,15 @@
   EXO_THUMBNAIL_SIZE_LARGE  = 256,
 } ExoThumbnailSize;
 
+G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT 
 GdkPixbuf *_exo_thumbnail_get_for_file (const gchar     *filename,
                                         ExoThumbnailSize size,
-                                        GError         **error) G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+                                        GError         **error);
+G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT 
 GdkPixbuf *_exo_thumbnail_get_for_uri  (const gchar     *uri,
                                         ExoThumbnailSize size,
-                                        GError         **error) G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+                                        GError         **error);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_THUMBNAIL_H__ */
--- exo-0.3.4/exo/exo-wrap-table.h.fixgccism	2007-12-09 20:05:00.799837118 +0100
+++ exo-0.3.4/exo/exo-wrap-table.h	2007-12-09 20:09:01.454644360 +0100
@@ -28,7 +28,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoWrapTablePrivate ExoWrapTablePrivate;
 typedef struct _ExoWrapTableClass   ExoWrapTableClass;
@@ -60,9 +60,9 @@
   ExoWrapTablePrivate *priv;
 };
 
-GType      exo_wrap_table_get_type        (void) G_GNUC_CONST;
+G_GNUC_CONST GType      exo_wrap_table_get_type        (void);
 
-GtkWidget *exo_wrap_table_new             (gboolean            homogeneous) G_GNUC_MALLOC;
+G_GNUC_MALLOC GtkWidget *exo_wrap_table_new             (gboolean            homogeneous);
 
 guint      exo_wrap_table_get_col_spacing (const ExoWrapTable *table);
 void       exo_wrap_table_set_col_spacing (ExoWrapTable       *table,
@@ -76,6 +76,6 @@
 void       exo_wrap_table_set_homogeneous (ExoWrapTable       *table,
                                            gboolean            homogeneous);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_WRAP_TABLE_H__ */
--- exo-0.3.4/exo/exo-string.h.fixgccism	2007-12-09 21:14:58.016469954 +0100
+++ exo-0.3.4/exo/exo-string.h	2007-12-09 21:16:45.315442609 +0100
@@ -29,22 +29,26 @@
 
 #include <exo/exo-config.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
-gchar                *exo_str_elide_underscores  (const gchar     *text) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT
+gchar                *exo_str_elide_underscores  (const gchar     *text);
 
 gboolean              exo_str_is_equal           (const gchar     *a,
                                                   const gchar     *b);
 
+G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT
 gchar                *exo_str_replace            (const gchar     *str,
                                                   const gchar     *pattern,
-                                                  const gchar     *replacement) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+                                                  const gchar     *replacement);
 
+G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT
 gchar                *exo_strdup_strftime        (const gchar     *format,
-                                                  const struct tm *tm) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+                                                  const struct tm *tm);
 
+G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT
 gchar               **exo_strndupv               (gchar          **strv,
-                                                  gint             num) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+                                                  gint             num);
 
 G_CONST_RETURN gchar *exo_intern_string          (const gchar     *string);
 G_CONST_RETURN gchar *exo_intern_static_string   (const gchar     *string);
@@ -55,6 +59,6 @@
 #define I_(string) (exo_intern_static_string ((string)))
 #endif
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_STRING_H__ */
--- exo-0.3.4/exo/exo-toolbars-private.h.fixgccism	2007-12-09 19:50:21.889054608 +0100
+++ exo-0.3.4/exo/exo-toolbars-private.h	2007-12-09 19:57:53.923563359 +0100
@@ -30,17 +30,17 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
-GdkPixbuf *_exo_toolbars_new_separator_pixbuf (void) G_GNUC_INTERNAL G_GNUC_MALLOC;
-GtkWidget *_exo_toolbars_new_separator_image  (void) G_GNUC_INTERNAL G_GNUC_MALLOC;
+G_GNUC_INTERNAL G_GNUC_MALLOC GdkPixbuf *_exo_toolbars_new_separator_pixbuf (void);
+G_GNUC_INTERNAL G_GNUC_MALLOC GtkWidget *_exo_toolbars_new_separator_image  (void);
 
-GtkAction *_exo_toolbars_find_action          (GtkUIManager *ui_manager,
-                                               const gchar  *name) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL GtkAction *_exo_toolbars_find_action          (GtkUIManager *ui_manager,
+                                               const gchar  *name);
 
-void       _exo_toolbars_set_drag_cursor      (GtkWidget    *widget) G_GNUC_INTERNAL;
-void       _exo_toolbars_unset_drag_cursor    (GtkWidget    *widget) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL void       _exo_toolbars_set_drag_cursor      (GtkWidget    *widget);
+G_GNUC_INTERNAL void       _exo_toolbars_unset_drag_cursor    (GtkWidget    *widget);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_TOOLBARS_PRIVATE_H__ */
--- exo-0.3.4/exo/exo-icon-chooser-dialog.h.fixgccism	2007-12-09 19:13:37.025967750 +0100
+++ exo-0.3.4/exo/exo-icon-chooser-dialog.h	2007-12-09 19:14:04.208938133 +0100
@@ -29,7 +29,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoIconChooserDialogPrivate ExoIconChooserDialogPrivate;
 typedef struct _ExoIconChooserDialogClass   ExoIconChooserDialogClass;
@@ -73,6 +73,6 @@
 gboolean   exo_icon_chooser_dialog_set_icon (ExoIconChooserDialog *icon_chooser_dialog,
                                              const gchar          *icon);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_ICON_CHOOSER_DIALOG_H__ */
--- exo-0.3.4/exo/exo-url.h.fixgccism	2007-12-09 20:00:49.079399396 +0100
+++ exo-0.3.4/exo/exo-url.h	2007-12-09 20:02:21.647807098 +0100
@@ -27,10 +27,10 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #define EXO_URL_ERROR (exo_url_error_quark ())
-GQuark exo_url_error_quark (void) G_GNUC_CONST;
+G_GNUC_CONST GQuark exo_url_error_quark (void);
 
 /**
  * ExoUrlError:
@@ -57,6 +57,6 @@
                                      const gchar    *link,
                                      gpointer        user_data);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_URL_H__ */
--- exo-0.3.4/exo/exo-private.h.fixgccism	2007-12-09 18:22:20.724332835 +0100
+++ exo-0.3.4/exo/exo-private.h	2007-12-09 21:21:53.435727509 +0100
@@ -31,7 +31,7 @@
 
 #include <glib/gi18n-lib.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 /* support macros for debugging */
 #ifndef NDEBUG
@@ -85,22 +85,25 @@
 #define g_value_get_object(v)   (((const GValue *) (v))->data[0].v_pointer)
 #endif
 
-void  _exo_i18n_init                    (void) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL void  _exo_i18n_init                    (void);
 
+G_GNUC_INTERNAL 
 void  _exo_gtk_widget_send_focus_change (GtkWidget         *widget,
-                                         gboolean           in) G_GNUC_INTERNAL;
+                                         gboolean           in);
 
+G_GNUC_INTERNAL 
 GType _exo_g_type_register_simple       (GType              type_parent,
                                          const gchar       *type_name_static,
                                          guint              class_size,
                                          gpointer           class_init,
                                          guint              instance_size,
-                                         gpointer           instance_init) G_GNUC_INTERNAL;
+                                         gpointer           instance_init);
 
+G_GNUC_INTERNAL 
 void  _exo_g_type_add_interface_simple  (GType              instance_type,
                                          GType              interface_type,
-                                         GInterfaceInitFunc interface_init_func) G_GNUC_INTERNAL;
+                                         GInterfaceInitFunc interface_init_func);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_PRIVATE_H__ */
--- exo-0.3.4/exo/exo-thumbnail-preview.h.fixgccism	2007-12-09 19:36:43.507551931 +0100
+++ exo-0.3.4/exo/exo-thumbnail-preview.h	2007-12-09 19:38:27.891854284 +0100
@@ -29,7 +29,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoThumbnailPreviewClass ExoThumbnailPreviewClass;
 typedef struct _ExoThumbnailPreview      ExoThumbnailPreview;
@@ -41,13 +41,13 @@
 #define EXO_IS_THUMBNAIL_PREVIEW_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), EXO_TYPE_THUMBNAIL_PREVIEW))
 #define EXO_THUMBNAIL_PREVIEW_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS ((obj), EXO_TYPE_THUMBNAIL_PREVIEW, ExoThumbnailPreviewClass))
 
-GType      _exo_thumbnail_preview_get_type (void) G_GNUC_CONST G_GNUC_INTERNAL;
+G_GNUC_INTERNAL GType      _exo_thumbnail_preview_get_type (void) G_GNUC_CONST;
 
-GtkWidget *_exo_thumbnail_preview_new      (void) G_GNUC_INTERNAL G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
+G_GNUC_INTERNAL GtkWidget *_exo_thumbnail_preview_new      (void) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
 
-void       _exo_thumbnail_preview_set_uri  (ExoThumbnailPreview *thumbnail_preview,
-                                            const gchar         *uri) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL void       _exo_thumbnail_preview_set_uri  (ExoThumbnailPreview *thumbnail_preview,
+                                            const gchar         *uri);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_THUMBNAIL_PREVIEW_H__ */
--- exo-0.3.4/exo/exo-toolbars-editor-dialog.h.fixgccism	2007-12-09 19:45:27.085156784 +0100
+++ exo-0.3.4/exo/exo-toolbars-editor-dialog.h	2007-12-09 19:46:40.007969677 +0100
@@ -30,7 +30,7 @@
 
 #include <exo/exo-toolbars-model.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #define EXO_TYPE_TOOLBARS_EDITOR_DIALOG             (exo_toolbars_editor_dialog_get_type ())
 #define EXO_TOOLBARS_EDITOR_DIALOG(obj)             (G_TYPE_CHECK_INSTANCE_CAST ((obj), EXO_TYPE_TOOLBARS_EDITOR_DIALOG, ExoToolbarsEditorDialog))
@@ -60,11 +60,11 @@
 };
 
 
-GType      exo_toolbars_editor_dialog_get_type       (void) G_GNUC_CONST;
+G_GNUC_CONST GType      exo_toolbars_editor_dialog_get_type       (void);
 
 GtkWidget *exo_toolbars_editor_dialog_new_with_model (GtkUIManager     *ui_manager,
                                                       ExoToolbarsModel *model);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_TOOLBARS_EDITOR_DIALOG_H__ */
--- exo-0.3.4/exo/exo-gdk-pixbuf-extensions.h.fixgccism	2007-12-09 19:09:29.117186310 +0100
+++ exo-0.3.4/exo/exo-gdk-pixbuf-extensions.h	2007-12-09 19:10:15.187547308 +0100
@@ -31,7 +31,7 @@
 
 #include <gdk/gdk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 GdkPixbuf *exo_gdk_pixbuf_colorize                  (const GdkPixbuf *src,
                                                      const GdkColor  *color) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
@@ -62,7 +62,7 @@
                                                      gboolean         preserve_aspect_ratio,
                                                      GError         **error) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_GDK_PIXBUF_EXTENSIONS_H__ */
 
--- exo-0.3.4/exo/exo-toolbars-view.h.fixgccism	2007-12-09 19:58:18.476754195 +0100
+++ exo-0.3.4/exo/exo-toolbars-view.h	2007-12-09 19:58:42.704437495 +0100
@@ -30,7 +30,7 @@
 
 #include <exo/exo-toolbars-model.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #define EXO_TYPE_TOOLBARS_VIEW            (exo_toolbars_view_get_type ())
 #define EXO_TOOLBARS_VIEW(obj)            (G_TYPE_CHECK_INSTANCE_CAST ((obj), EXO_TYPE_TOOLBARS_VIEW, ExoToolbarsView))
@@ -84,6 +84,6 @@
 void              exo_toolbars_view_set_ui_manager  (ExoToolbarsView      *view,
                                                      GtkUIManager         *ui_manager);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_TOOLBARS_VIEW_H__ */
--- exo-0.3.4/exo/exo-execute.h.fixgccism	2007-12-09 19:00:34.694333523 +0100
+++ exo-0.3.4/exo/exo-execute.h	2007-12-09 21:22:30.033442445 +0100
@@ -27,7 +27,7 @@
 
 #include <gdk/gdk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 gboolean exo_execute_preferred_application            (const gchar *category,
                                                        const gchar *parameter,
@@ -51,6 +51,6 @@
                                                        GdkScreen   *screen,
                                                        GError     **error);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_EXECUTE_H__ */
--- exo-0.3.4/exo/exo-cell-renderer-icon.h.fixgccism	2007-12-09 18:55:52.147410842 +0100
+++ exo-0.3.4/exo/exo-cell-renderer-icon.h	2007-12-09 18:56:24.820278026 +0100
@@ -29,7 +29,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoCellRendererIconPrivate ExoCellRendererIconPrivate;
 typedef struct _ExoCellRendererIconClass   ExoCellRendererIconClass;
@@ -66,6 +66,6 @@
 
 GtkCellRenderer *exo_cell_renderer_icon_new      (void) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_CELL_RENDERER_ICON_H__ */
--- exo-0.3.4/exo/exo-mount-point.h.fixgccism	2007-12-09 19:19:44.712696573 +0100
+++ exo-0.3.4/exo/exo-mount-point.h	2007-12-09 19:19:59.727252634 +0100
@@ -27,7 +27,7 @@
 
 #include <exo/exo-config.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 /**
  * ExoMountPointFlags:
@@ -128,6 +128,6 @@
  **/
 #define exo_mount_point_list_configured(error) (exo_mount_point_list_matched (EXO_MOUNT_POINT_MATCH_CONFIGURED, NULL, NULL, NULL, (error)))
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_MOUNT_POINT_H__ */
--- exo-0.3.4/exo/exo-icon-chooser-model.h.fixgccism	2007-12-09 19:14:25.366771315 +0100
+++ exo-0.3.4/exo/exo-icon-chooser-model.h	2007-12-09 19:16:55.794778190 +0100
@@ -29,7 +29,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoIconChooserModelClass ExoIconChooserModelClass;
 typedef struct _ExoIconChooserModel      ExoIconChooserModel;
@@ -86,15 +86,15 @@
   EXO_ICON_CHOOSER_MODEL_N_COLUMNS,
 } ExoIconChooserModelColumn;
 
-GType                _exo_icon_chooser_model_get_type               (void) G_GNUC_CONST G_GNUC_INTERNAL;
+G_GNUC_INTERNAL GType                _exo_icon_chooser_model_get_type               (void) G_GNUC_CONST;
 
-ExoIconChooserModel *_exo_icon_chooser_model_get_for_widget         (GtkWidget           *widget) G_GNUC_INTERNAL G_GNUC_WARN_UNUSED_RESULT;
-ExoIconChooserModel *_exo_icon_chooser_model_get_for_icon_theme     (GtkIconTheme        *icon_theme) G_GNUC_INTERNAL G_GNUC_WARN_UNUSED_RESULT;
+G_GNUC_INTERNAL ExoIconChooserModel *_exo_icon_chooser_model_get_for_widget         (GtkWidget           *widget) G_GNUC_WARN_UNUSED_RESULT;
+G_GNUC_INTERNAL ExoIconChooserModel *_exo_icon_chooser_model_get_for_icon_theme     (GtkIconTheme        *icon_theme) G_GNUC_WARN_UNUSED_RESULT;
 
-gboolean             _exo_icon_chooser_model_get_iter_for_icon_name (ExoIconChooserModel *model,
+G_GNUC_INTERNAL gboolean             _exo_icon_chooser_model_get_iter_for_icon_name (ExoIconChooserModel *model,
                                                                      GtkTreeIter         *iter,
-                                                                     const gchar         *icon_name) G_GNUC_INTERNAL;
+                                                                     const gchar         *icon_name);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_ICON_CHOOSER_MODEL_H__ */
--- exo-0.3.4/exo/exo-cell-renderer-ellipsized-text.h.fixgccism	2007-12-09 18:53:44.869771430 +0100
+++ exo-0.3.4/exo/exo-cell-renderer-ellipsized-text.h	2007-12-09 18:54:22.756008134 +0100
@@ -31,7 +31,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoCellRendererEllipsizedTextPrivate ExoCellRendererEllipsizedTextPrivate;
 typedef struct _ExoCellRendererEllipsizedTextClass   ExoCellRendererEllipsizedTextClass;
@@ -61,6 +61,6 @@
 
 GtkCellRenderer *exo_cell_renderer_ellipsized_text_new      (void) G_GNUC_MALLOC G_GNUC_WARN_UNUSED_RESULT;
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_CELL_RENDERER_ELLIPSIZED_TEXT_H__ */
--- exo-0.3.4/exo/exo-tree-view.h.fixgccism	2007-12-09 19:59:03.397304937 +0100
+++ exo-0.3.4/exo/exo-tree-view.h	2007-12-09 20:00:13.389093783 +0100
@@ -27,7 +27,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoTreeViewPrivate ExoTreeViewPrivate;
 typedef struct _ExoTreeViewClass   ExoTreeViewClass;
@@ -65,9 +65,9 @@
   ExoTreeViewPrivate *priv;
 };
 
-GType      exo_tree_view_get_type                 (void) G_GNUC_CONST;
+G_GNUC_CONST GType      exo_tree_view_get_type                 (void);
 
-GtkWidget *exo_tree_view_new                      (void) G_GNUC_MALLOC;
+G_GNUC_MALLOC GtkWidget *exo_tree_view_new                      (void);
 
 gboolean   exo_tree_view_get_single_click         (const ExoTreeView *tree_view);
 void       exo_tree_view_set_single_click         (ExoTreeView       *tree_view,
@@ -77,6 +77,6 @@
 void       exo_tree_view_set_single_click_timeout (ExoTreeView       *tree_view,
                                                    guint              single_click_timeout);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_TREE_VIEW_H__ */
--- exo-0.3.4/exo/exo-icon-view.h.fixgccism	2007-12-09 19:17:15.508450094 +0100
+++ exo-0.3.4/exo/exo-icon-view.h	2007-12-09 19:18:37.123497016 +0100
@@ -30,7 +30,7 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoIconViewPrivate    ExoIconViewPrivate;
 typedef struct _ExoIconViewClass      ExoIconViewClass;
@@ -341,6 +341,6 @@
                                                                        gpointer                      search_position_data,
                                                                        GDestroyNotify                search_position_destroy);
  
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* __EXO_ICON_VIEW_H__ */
--- exo-0.3.4/exo-helper/exo-helper-launcher-dialog.h.fixgccism	2007-12-09 20:44:46.504225894 +0100
+++ exo-0.3.4/exo-helper/exo-helper-launcher-dialog.h	2007-12-09 20:45:49.005639332 +0100
@@ -22,7 +22,7 @@
 
 #include <exo-helper/exo-helper.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoHelperLauncherDialogClass ExoHelperLauncherDialogClass;
 typedef struct _ExoHelperLauncherDialog      ExoHelperLauncherDialog;
@@ -34,14 +34,14 @@
 #define EXO_IS_HELPER_LAUNCHER_DIALOG_CLASS(klass)  (G_TYPE_CHECK_CLASS_TYPE ((klass), EXO_TYPE_HELPER_LAUNCHER_DIALOG))
 #define EXO_HELPER_LAUNCHER_DIALOG_GET_CLASS(obj)   (G_TYPE_INSTANCE_GET_CLASS ((obj), EXOT_TYPE_HELPER_LAUNCHER_DIALOg, ExoHelperLauncherDialogClass))
 
-GType             exo_helper_launcher_dialog_get_type     (void) G_GNUC_CONST;
+G_GNUC_CONST GType             exo_helper_launcher_dialog_get_type     (void);
 
-GtkWidget        *exo_helper_launcher_dialog_new          (ExoHelperCategory              category) G_GNUC_MALLOC;
+G_GNUC_MALLOC GtkWidget        *exo_helper_launcher_dialog_new          (ExoHelperCategory              category);
 
 ExoHelperCategory exo_helper_launcher_dialog_get_category (const ExoHelperLauncherDialog *launcher_dialog);
 void              exo_helper_launcher_dialog_set_category (ExoHelperLauncherDialog       *launcher_dialog,
                                                            ExoHelperCategory              category);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_HELPER_LAUNCHER_DIALOG_H__ */
--- exo-0.3.4/exo-helper/exo-helper-chooser-dialog.h.fixgccism	2007-12-09 20:37:49.396700190 +0100
+++ exo-0.3.4/exo-helper/exo-helper-chooser-dialog.h	2007-12-09 20:38:43.546300466 +0100
@@ -22,7 +22,7 @@
 
 #include <exo-helper/exo-helper-chooser.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoHelperChooserDialogClass ExoHelperChooserDialogClass;
 typedef struct _ExoHelperChooserDialog      ExoHelperChooserDialog;
@@ -34,10 +34,10 @@
 #define EXO_IS_HELPER_CHOOSER_DIALOG_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), EXO_TYPE_HELPER_CHOOSER_DIALOG))
 #define EXO_HELPER_CHOOSER_DIALOG_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS ((obj), EXO_TYPE_HELPER_CHOOSER_DIALOG, ExoHelperChooserDialogClass))
 
-GType      exo_helper_chooser_dialog_get_type (void) G_GNUC_CONST;
+G_GNUC_CONST GType      exo_helper_chooser_dialog_get_type (void);
 
-GtkWidget *exo_helper_chooser_dialog_new      (void) G_GNUC_MALLOC;
+G_GNUC_MALLOC GtkWidget *exo_helper_chooser_dialog_new      (void);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_HELPER_CHOOSER_DIALOG_H__ */
--- exo-0.3.4/exo-helper/exo-helper-utils.h.fixgccism	2007-12-09 20:46:12.034293646 +0100
+++ exo-0.3.4/exo-helper/exo-helper-utils.h	2007-12-09 20:47:18.838082648 +0100
@@ -22,12 +22,12 @@
 
 #include <exo-helper/exo-helper.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
-gboolean exo_helper_category_from_string (const gchar       *string,
-                                          ExoHelperCategory *category_return) G_GNUC_INTERNAL;
-gchar   *exo_helper_category_to_string   (ExoHelperCategory  category) G_GNUC_INTERNAL G_GNUC_MALLOC;
+G_GNUC_INTERNAL gboolean exo_helper_category_from_string (const gchar       *string,
+                                          ExoHelperCategory *category_return);
+G_GNUC_INTERNAL G_GNUC_MALLOC gchar   *exo_helper_category_to_string   (ExoHelperCategory  category);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_HELPER_UTILS_H__ */
--- exo-0.3.4/exo-helper/exo-preferred-applications-settings.c.fixgccism	2007-12-09 20:49:29.216118519 +0100
+++ exo-0.3.4/exo-helper/exo-preferred-applications-settings.c	2007-12-09 20:50:10.922253286 +0100
@@ -77,4 +77,4 @@
 
 
 
-MCS_PLUGIN_CHECK_INIT;
+MCS_PLUGIN_CHECK_INIT
--- exo-0.3.4/exo-helper/exo-helper-chooser.h.fixgccism	2007-12-09 20:41:42.569371890 +0100
+++ exo-0.3.4/exo-helper/exo-helper-chooser.h	2007-12-09 20:44:17.398415940 +0100
@@ -22,7 +22,7 @@
 
 #include <exo-helper/exo-helper.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #define EXO_TYPE_HELPER_CHOOSER            (exo_helper_chooser_get_type ())
 #define EXO_HELPER_CHOOSER(obj)            (G_TYPE_CHECK_INSTANCE_CAST ((obj), EXO_TYPE_HELPER_CHOOSER, ExoHelperChooser))
@@ -34,9 +34,9 @@
 typedef struct _ExoHelperChooserClass ExoHelperChooserClass;
 typedef struct _ExoHelperChooser      ExoHelperChooser;
 
-GType              exo_helper_chooser_get_type      (void) G_GNUC_CONST;
+G_GNUC_CONST GType              exo_helper_chooser_get_type      (void);
 
-GtkWidget         *exo_helper_chooser_new           (ExoHelperCategory       category) G_GNUC_MALLOC;
+G_GNUC_MALLOC GtkWidget         *exo_helper_chooser_new           (ExoHelperCategory       category);
 
 ExoHelperCategory  exo_helper_chooser_get_category  (const ExoHelperChooser *chooser);
 void               exo_helper_chooser_set_category  (ExoHelperChooser       *chooser,
@@ -44,6 +44,6 @@
 
 gboolean           exo_helper_chooser_get_is_valid  (const ExoHelperChooser *chooser);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_HELPER_CHOOSER_H__ */
--- exo-0.3.4/exo-helper/exo-helper.h.fixgccism	2007-12-09 20:47:48.158783810 +0100
+++ exo-0.3.4/exo-helper/exo-helper.h	2007-12-09 20:48:57.084040763 +0100
@@ -22,7 +22,7 @@
 
 #include <exo/exo.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef enum /*< enum,prefix=EXO_HELPER >*/
 {
@@ -42,7 +42,7 @@
 #define EXO_IS_HELPER_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), EXO_TYPE_HELPER))
 #define EXO_HELPER_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS ((obj), EXO_TYPE_HELPER, ExoHelperClass))
 
-GType              exo_helper_get_type      (void) G_GNUC_CONST;
+G_GNUC_CONST GType              exo_helper_get_type      (void);
 ExoHelperCategory  exo_helper_get_category  (const ExoHelper   *helper);
 const gchar       *exo_helper_get_id        (const ExoHelper   *helper);
 const gchar       *exo_helper_get_name      (const ExoHelper   *helper);
@@ -64,7 +64,7 @@
 typedef struct _ExoHelperDatabaseClass ExoHelperDatabaseClass;
 typedef struct _ExoHelperDatabase      ExoHelperDatabase;
 
-GType               exo_helper_database_get_type        (void) G_GNUC_CONST;
+G_GNUC_CONST GType               exo_helper_database_get_type        (void);
 ExoHelperDatabase  *exo_helper_database_get             (void);
 ExoHelper          *exo_helper_database_get_default     (ExoHelperDatabase *database,
                                                          ExoHelperCategory  category);
@@ -80,6 +80,6 @@
                                                          ExoHelperCategory  category,
                                                          const gchar       *command);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_HELPER_H__ */
--- exo-0.3.4/python/exo.override.fixgccism	2007-12-09 21:09:14.235683415 +0100
+++ exo-0.3.4/python/exo.override	2007-12-09 21:10:51.571711518 +0100
@@ -28,9 +28,9 @@
 #include "pyexo.h"
 
 /* mark internal symbols with G_GNUC_INTERNAL */
-void exo_add_constants    (PyObject    *module,
-                           const gchar *strip_prefix) G_GNUC_INTERNAL;
-void exo_register_classes (PyObject    *d) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL void exo_add_constants    (PyObject    *module,
+                           const gchar *strip_prefix);
+G_GNUC_INTERNAL void exo_register_classes (PyObject    *d);
 
 %%
 modulename exo
--- exo-0.3.4/python/exomodule.c.fixgccism	2007-12-09 21:11:03.242993333 +0100
+++ exo-0.3.4/python/exomodule.c	2007-12-09 21:11:51.495253961 +0100
@@ -29,9 +29,9 @@
 
 
 /* mark internal symbols with G_GNUC_INTERNAL */
-void exo_add_constants    (PyObject    *module,
-                           const gchar *strip_prefix) G_GNUC_INTERNAL;
-void exo_register_classes (PyObject    *d) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL void exo_add_constants    (PyObject    *module,
+                           const gchar *strip_prefix);
+G_GNUC_INTERNAL void exo_register_classes (PyObject    *d);
 
 extern PyMethodDef exo_functions[];
 
--- exo-0.3.4/python/pyexo.h.fixgccism	2007-12-09 21:12:15.569906821 +0100
+++ exo-0.3.4/python/pyexo.h	2007-12-09 21:13:06.688095272 +0100
@@ -28,7 +28,7 @@
 
 #include <exo/exo.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 /* ---------- ExoBinding ---------- */
 typedef struct
@@ -53,13 +53,13 @@
 
 /* ---------- PyGTK helpers ---------- */
 #if !defined(pygtk_tree_path_from_pyobject)
-GtkTreePath *pygtk_tree_path_from_pyobject (PyObject *object) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL GtkTreePath *pygtk_tree_path_from_pyobject (PyObject *object);
 #endif
 
 #if !defined(pygtk_tree_path_to_pyobject)
-PyObject *pygtk_tree_path_to_pyobject (GtkTreePath *path) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL PyObject *pygtk_tree_path_to_pyobject (GtkTreePath *path);
 #endif
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__PYEXO_H__ */
--- exo-0.3.4/exo-desktop-item-edit/exo-die-utils.h.fixgccism	2007-12-09 20:30:21.886517293 +0100
+++ exo-0.3.4/exo-desktop-item-edit/exo-die-utils.h	2007-12-09 20:31:48.176320284 +0100
@@ -23,18 +23,18 @@
 
 #include <exo/exo.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
-void      exo_die_g_key_file_set_locale_value (GKeyFile    *key_file,
+G_GNUC_INTERNAL void      exo_die_g_key_file_set_locale_value (GKeyFile    *key_file,
                                                const gchar *group,
                                                const gchar *key,
-                                               const gchar *value) G_GNUC_INTERNAL;
+                                               const gchar *value);
 
-gboolean  exo_die_g_key_file_save             (GKeyFile    *key_file,
+G_GNUC_INTERNAL gboolean  exo_die_g_key_file_save             (GKeyFile    *key_file,
                                                gboolean     create,
                                                const gchar *base,
-                                               GError     **error) G_GNUC_INTERNAL;
+                                               GError     **error);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_DIE_UTILS_H__ */
--- exo-0.3.4/exo-desktop-item-edit/exo-die-enum-types.h.fixgccism	2007-12-09 20:25:16.199519757 +0100
+++ exo-0.3.4/exo-desktop-item-edit/exo-die-enum-types.h	2007-12-09 20:25:58.745939428 +0100
@@ -23,7 +23,7 @@
 
 #include <exo/exo.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 #define EXO_DIE_TYPE_EDITOR_MODE (exo_die_editor_mode_get_type ())
 
@@ -38,8 +38,8 @@
   EXO_DIE_EDITOR_MODE_LINK,
 } ExoDieEditorMode;
 
-GType exo_die_editor_mode_get_type (void) G_GNUC_CONST;
+G_GNUC_CONST GType exo_die_editor_mode_get_type (void);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_DIE_ENUM_TYPES_H__ */
--- exo-0.3.4/exo-desktop-item-edit/exo-die-desktop-model.h.fixgccism	2007-12-09 20:14:58.795224924 +0100
+++ exo-0.3.4/exo-desktop-item-edit/exo-die-desktop-model.h	2007-12-09 20:16:47.080171218 +0100
@@ -23,7 +23,7 @@
 
 #include <exo/exo.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoDieDesktopModelClass ExoDieDesktopModelClass;
 typedef struct _ExoDieDesktopModel      ExoDieDesktopModel;
@@ -59,15 +59,15 @@
   EXO_DIE_DESKTOP_MODEL_N_COLUMNS,
 } ExoDieDesktopModelColumn;
 
-GType               exo_die_desktop_model_get_type    (void) G_GNUC_CONST;
+G_GNUC_CONST GType               exo_die_desktop_model_get_type    (void);
 
-ExoDieDesktopModel *exo_die_desktop_model_new         (void) G_GNUC_MALLOC;
+G_GNUC_MALLOC ExoDieDesktopModel *exo_die_desktop_model_new         (void);
 
 gboolean            exo_die_desktop_model_match_func  (GtkEntryCompletion *completion,
                                                        const gchar        *key,
                                                        GtkTreeIter        *iter,
                                                        gpointer            user_data);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_DIE_DESKTOP_MODEL_H__ */
--- exo-0.3.4/exo-desktop-item-edit/exo-die-command-model.h.fixgccism	2007-12-09 20:12:57.479759855 +0100
+++ exo-0.3.4/exo-desktop-item-edit/exo-die-command-model.h	2007-12-09 20:14:42.403857551 +0100
@@ -23,7 +23,7 @@
 
 #include <exo/exo.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoDieCommandModelClass ExoDieCommandModelClass;
 typedef struct _ExoDieCommandModel      ExoDieCommandModel;
@@ -47,10 +47,10 @@
   EXO_DIE_COMMAND_MODEL_N_COLUMNS,
 } ExoDieCommandModelColumn;
 
-GType               exo_die_command_model_get_type (void) G_GNUC_CONST;
+G_GNUC_CONST GType               exo_die_command_model_get_type (void);
 
-ExoDieCommandModel *exo_die_command_model_new      (void) G_GNUC_MALLOC;
+G_GNUC_MALLOC ExoDieCommandModel *exo_die_command_model_new      (void);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_DIE_COMMAND_MODEL_H__ */
--- exo-0.3.4/exo-desktop-item-edit/exo-die-command-entry.h.fixgccism	2007-12-09 20:11:36.665400934 +0100
+++ exo-0.3.4/exo-desktop-item-edit/exo-die-command-entry.h	2007-12-09 20:12:35.957807665 +0100
@@ -23,7 +23,7 @@
 
 #include <exo/exo.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoDieCommandEntryClass ExoDieCommandEntryClass;
 typedef struct _ExoDieCommandEntry      ExoDieCommandEntry;
@@ -35,14 +35,14 @@
 #define EXO_DIE_IS_COMMAND_ENTRY_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), EXO_DIE_TYPE_COMMAND_ENTRY))
 #define EXO_DIE_COMMAND_ENTRY_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS ((obj), EXO_DIE_TYPE_COMMAND_ENTRY, ExoDieCommandEntryClass))
 
-GType        exo_die_command_entry_get_type (void) G_GNUC_CONST;
+G_GNUC_CONST GType        exo_die_command_entry_get_type (void);
 
-GtkWidget   *exo_die_command_entry_new      (void) G_GNUC_MALLOC;
+G_GNUC_MALLOC GtkWidget   *exo_die_command_entry_new      (void);
 
 const gchar *exo_die_command_entry_get_text (ExoDieCommandEntry *command_entry);
 void         exo_die_command_entry_set_text (ExoDieCommandEntry *command_entry,
                                              const gchar        *text);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_DIE_COMMAND_ENTRY_H__ */
--- exo-0.3.4/exo-desktop-item-edit/exo-die-editor.h.fixgccism	2007-12-09 20:23:53.326516840 +0100
+++ exo-0.3.4/exo-desktop-item-edit/exo-die-editor.h	2007-12-09 20:24:50.688386622 +0100
@@ -23,7 +23,7 @@
 
 #include <exo-desktop-item-edit/exo-die-enum-types.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _ExoDieEditorClass ExoDieEditorClass;
 typedef struct _ExoDieEditor      ExoDieEditor;
@@ -35,9 +35,9 @@
 #define EXO_DIE_IS_EDITOR_CLASS(klass)  (G_TYPE_CHECK_CLASS_TYPE ((klass), EXO_DIE_TYPE_EDITOR))
 #define EXO_DIE_EDITOR_GET_CLASS(obj)   (G_TYPE_INSTANCE_GET_CLASS ((obj), EXO_DIE_TYPE_EDITOR, ExoDieEditorClass))
 
-GType             exo_die_editor_get_type     (void) G_GNUC_CONST;
+G_GNUC_CONST GType             exo_die_editor_get_type     (void);
 
-GtkWidget        *exo_die_editor_new          (void) G_GNUC_MALLOC;
+G_GNUC_MALLOC GtkWidget        *exo_die_editor_new          (void);
 
 gboolean          exo_die_editor_get_complete (ExoDieEditor     *editor);
 
@@ -73,6 +73,6 @@
 void              exo_die_editor_set_terminal (ExoDieEditor     *editor,
                                                gboolean          terminal);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__EXO_DIE_EDITOR_H__ */
