--- mediastreamer-2.2.4/include/mediastreamer2/rfc3984.h.orig	2008-09-11 02:21:38.000000000 +0530
+++ mediastreamer-2.2.4/include/mediastreamer2/rfc3984.h	2009-05-14 17:33:39.822889059 +0530
@@ -29,7 +29,7 @@
 
 typedef struct Rfc3984Context{
 	MSQueue q;
-	mblk_t *m;
+	ortp_mblk_t *m;
 	int maxsz;
 	uint32_t last_ts;
 	uint8_t mode;
@@ -48,7 +48,7 @@
 void rfc3984_pack(Rfc3984Context *ctx, MSQueue *naluq, MSQueue *rtpq, uint32_t ts);
 
 /*process incoming rtp data and output NALUs, whenever possible*/
-void rfc3984_unpack(Rfc3984Context *ctx, mblk_t *im, MSQueue *naluq);
+void rfc3984_unpack(Rfc3984Context *ctx, ortp_mblk_t *im, MSQueue *naluq);
 
 void rfc3984_uninit(Rfc3984Context *ctx);
 
