--- gani-2.6.2/adddrv.sh.orig	2009-01-31 00:16:11.000000000 +0530
+++ gani-2.6.2/adddrv.sh	2009-04-04 14:05:45.514039834 +0530
@@ -11,29 +11,5 @@
 set DEVLIST = ( $DEVLIST '"pciex10ec,8168"' )	# PCI-E 8168/811x
 set DEVLIST = ( $DEVLIST '"pciex10ec,8136"' )	# PCI-E 810xE
 
-#echo $DEVLIST
-
-set DEVLIST2 = ( )
-foreach i ($DEVLIST)
-	set pcidev = `grep $i /etc/driver_aliases`
-	echo $pcidev
-	if ("$pcidev" == "") then
-		set DEVLIST2 = ( $DEVLIST2 "$i" )
-	endif
-end
-
-#echo $DEVLIST2
-if ("$DEVLIST2" == "") then
-	echo nothing to do.
-	exit 1
-endif
-
-set existing = `grep "gani " /etc/driver_aliases`
-#echo $existing
-
-if ("$existing" == "") then
-	/usr/sbin/add_drv -n -v -m '* 0600 root sys' -i "$DEVLIST2" gani
-else 
-	/usr/sbin/update_drv -a -v -m '* 0600 root sys' -i "$DEVLIST2" gani
-endif
+/usr/sbin/add_drv -n -v -m '* 0600 root sys' -i "$DEVLIST" gani
 sync
