--- ortp-0.13.1.orig/include/ortp/telephonyevents.h	2007-02-14 08:33:11.000000000 -0500
+++ ortp-0.13.1/include/ortp/telephonyevents.h	2008-05-16 19:50:47.000000000 -0400
@@ -61,19 +61,19 @@
 int rtp_session_send_telephone_events_supported(RtpSession *session);
 int rtp_session_recv_telephone_events_supported(RtpSession *session);
 
-mblk_t	*rtp_session_create_telephone_event_packet(RtpSession *session, int start);
+ortp_mblk_t	*rtp_session_create_telephone_event_packet(RtpSession *session, int start);
 
 int rtp_session_add_telephone_event(RtpSession *session,
-			mblk_t *packet, uint8_t event, int end, uint8_t volume, uint16_t duration);
+			ortp_mblk_t *packet, uint8_t event, int end, uint8_t volume, uint16_t duration);
 			
 int rtp_session_read_telephone_event(RtpSession *session,
-		mblk_t *packet,telephone_event_t **tab);
+		ortp_mblk_t *packet,telephone_event_t **tab);
 
 /* high level functions*/
 int rtp_session_send_dtmf(RtpSession *session, char dtmf, uint32_t userts);
 int rtp_session_send_dtmf2(RtpSession *session, char dtmf, uint32_t userts, int duration);
 /* for high level telephony event callback */
-void rtp_session_check_telephone_events(RtpSession *session, mblk_t *m0);
+void rtp_session_check_telephone_events(RtpSession *session, ortp_mblk_t *m0);
 
 #ifdef __cplusplus
 }
