--- pciutils-3.1.2/lib/internal.h.orig	2009-05-19 22:54:47.064262320 +0530
+++ pciutils-3.1.2/lib/internal.h	2009-05-19 22:55:09.765511997 +0530
@@ -8,16 +8,9 @@
 
 #include "config.h"
 
-#ifdef PCI_SHARED_LIB
-#define PCI_ABI __attribute__((visibility("default")))
-#define STATIC_ALIAS(_decl, _for)
-#define DEFINE_ALIAS(_decl, _for) extern _decl __attribute__((alias(#_for)))
-#define SYMBOL_VERSION(_int, _ext) asm(".symver " #_int "," #_ext)
-#else
 #define STATIC_ALIAS(_decl, _for) _decl { return _for; }
 #define DEFINE_ALIAS(_decl, _for)
 #define SYMBOL_VERSION(_int, _ext)
-#endif
 
 #include "pci.h"
 #include "sysdep.h"
