--- qt-x11-opensource-src-4.5.2/src/phonon/phonon.pro.orig	2009-06-20 10:27:57.000000000 +0530
+++ qt-x11-opensource-src-4.5.2/src/phonon/phonon.pro	2009-08-29 19:23:56.659692300 +0530
@@ -1,4 +1,4 @@
-TARGET = phonon
+TARGET = QtPhonon
 include(../qbase.pri)
 
 PHONON_MAJOR_VERSION = $${QT_MAJOR_VERSION}
@@ -6,6 +6,8 @@
 PHONON_PATCH_VERSION = 1
 VERSION = $${PHONON_MAJOR_VERSION}.$${PHONON_MINOR_VERSION}.$${PHONON_PATCH_VERSION}
 
+DEPENDPATH += ../opengl
+DEPENDPATH += ../svg
 DEPENDPATH += .
 DEFINES += MAKE_PHONON_LIB
 
