--- qt-x11-opensource-src-4.5.2/mkspecs/features/qt.prf.orig	2009-06-20 10:27:53.000000000 +0530
+++ qt-x11-opensource-src-4.5.2/mkspecs/features/qt.prf	2009-08-29 19:27:08.712737719 +0530
@@ -149,7 +149,7 @@
    else:isEqual(QTLIB, scripttools):qlib = QtScriptTools
    else:isEqual(QTLIB, testlib):qlib = QtTest
    else:isEqual(QTLIB, dbus):qlib = QtDBus
-   else:isEqual(QTLIB, phonon):qlib = phonon
+   else:isEqual(QTLIB, phonon):qlib = QtPhonon
    else:isEqual(QTLIB, webkit):qlib = QtWebKit
    else:message("Unknown QT: $$QTLIB"):qlib =
    !isEmpty(qlib) {
