--- sqlite-3.6.7-old/Makefile.in      Thu Dec  4 13:41:45 2008
+++ sqlite-3.6.7/Makefile.in  Wed Feb  4 16:45:34 2009
@@ -468,9 +468,15 @@
 # all that automatic generation.
 #
 .target_source:	$(SRC)
-	rm -rf tsrc
 	mkdir -p tsrc
-	cp $(SRC) tsrc
+	for i in $(SRC); \
+	do \
+		if /bin/test $$i -nt tsrc/$$i; \
+		then \
+			cp $$i tsrc; \
+			touch target_source; \
+		fi; \
+	done
 	rm tsrc/sqlite.h.in tsrc/parse.y
 	touch .target_source
 
@@ -767,7 +773,7 @@
 clean:	
 	rm -f *.lo *.la *.o sqlite3$(TEXE) libsqlite3.la
 	rm -f sqlite3.h opcodes.*
-	rm -rf .libs .deps tsrc
+	rm -rf .libs .deps tsrc solaris
 	rm -f lemon$(BEXE) lempar.c parse.* sqlite*.tar.gz
 	rm -f mkkeywordhash$(BEXE) keywordhash.h
 	rm -f $(PUBLISH)
@@ -797,3 +803,30 @@
 sqlite3.dll: $(REAL_LIBOBJ) sqlite3.def
 	$(TCC) -shared -o $@ sqlite3.def \
 		-Wl,"--strip-all" $(REAL_LIBOBJ)
+ 
+solaris:
+	mkdir -p solaris
+
+solaris/libsqlite3.so.0: solaris sqlite3.c
+	rm -f solaris/libsqlite3.so.0
+	$(TCC) -DSQLITE_PRIVATE= -o solaris/libsqlite3.so.0 -G -KPIC sqlite3.c \
+		-L$(ROOT)/usr/lib  -M $(LIBSQLITE_MAPFILE) \
+		-h libsqlite3.so.0 -lc -lpthread $(TLIBS)
+
+solaris/libsqlite3.so: solaris/libsqlite3.so.0
+	rm -f solaris/libsqlite3.so
+	ln -s libsqlite3.so.0 solaris/libsqlite3.so
+ 
+solaris/sqlite3: solaris/libsqlite3.so
+	rm -f solaris/sqlite3
+	$(TCC) -o solaris/sqlite3 $(TOP)/src/shell.c \
+		-I$(ROOT)/usr/include -L$(TOP)/solaris -L$(ROOT)/usr/lib \
+		$(TLIBS) -M $(CMD_MAPFILE) -lsqlite3
+
+solaris/libtclsqlite3.so: solaris/libsqlite3.so
+	rm -f solaris/libtclsqlite3.so
+	$(TCC) -o solaris/libtclsqlite3.so -G -KPIC $(TOP)/src/tclsqlite.c \
+		-I$(ROOT)/usr/include -L$(TOP)/solaris -L$(ROOT)/usr/lib \
+		-h libtclsqlite3.so.0 -lsqlite3 -L/usr/lib -ltclstub8.4 \
+		-lpthread
+
