diff -up taglib-1.5/taglib/mpeg/id3v2/id3v2tag.cpp.kde#161721 taglib-1.5/taglib/mpeg/id3v2/id3v2tag.cpp
--- taglib-1.5/taglib/mpeg/id3v2/id3v2tag.cpp.kde#161721	2008-02-04 14:33:54.000000000 -0600
+++ taglib-1.5/taglib/mpeg/id3v2/id3v2tag.cpp	2008-10-06 15:52:08.000000000 -0500
@@ -125,7 +125,9 @@ String ID3v2::Tag::comment() const
 
   for(FrameList::ConstIterator it = comments.begin(); it != comments.end(); ++it)
   {
-    if(static_cast<CommentsFrame *>(*it)->description().isEmpty())
+    CommentsFrame *frame = dynamic_cast<CommentsFrame *>(*it);
+
+    if(frame && frame->description().isEmpty())
       return (*it)->toString();
   }
 
