--- xfce-mcs-manager-4.4.0-orig/xfce-mcs-manager/manager-plugin.h.in	Sat Jan 20 16:28:16 2007
+++ xfce-mcs-manager-4.4.0/xfce-mcs-manager/manager-plugin.h.in	Sun Mar  4 17:10:16 2007
@@ -26,7 +26,7 @@
 #include <gmodule.h>
 #include <libxfce4mcs/mcs-manager.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 /* Xfce4 mcs plugin API version */
 #define XFCE_MCS_PLUGIN_VERSION	@XFCE_MCS_PLUGIN_VERSION@
@@ -74,6 +74,6 @@
 /* prototype for plugin init function (should be implemented by plugin) */
 G_MODULE_EXPORT McsPluginInitResult mcs_plugin_init (McsPlugin *plugin);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !MANAGER_PLUGIN_H */
