--- zziplib-0.13.58/zzip/zip.c.orig	2009-08-23 17:08:22.000000000 +0530
+++ zziplib-0.13.58/zzip/zip.c	2009-11-14 22:53:26.371576472 +0530
@@ -88,7 +88,7 @@
                                 struct zzip_dir_hdr **hdr_return,
                                 zzip_plugin_io_t io);
 
-_zzip_inline char *__zzip_aligned4(char *p);
+char *__zzip_aligned4(char *p);
 
 /* ------------------------  harden routines ------------------------------ */
 
@@ -375,7 +375,7 @@
  * to pointer types but we do need only the lower bits in this implementation,
  * so we can just cast the value to a long value.
  */
-_zzip_inline char *
+char *
 __zzip_aligned4(char *p)
 {
 #define aligned4   __zzip_aligned4
