--- firefox3/modules/libpref/src/init/all.js.orig       2008-02-14 16:28:44.932870000 +0800
+++ firefox3/modules/libpref/src/init/all.js    2008-02-14 16:46:17.858131000 +0800
@@ -51,7 +51,8 @@
 pref("keyword.enabled", false);
 pref("general.useragent.locale", "chrome://global/locale/intl.properties");
 
-pref("general.config.obscure_value", 13); // for MCD .cfg files
+pref("general.config.filename","firefox.cfg");
+pref("general.config.obscure_value", 0); // for MCD .cfg files
 
 pref("general.warnOnAboutConfig", true);
