--- kdeadmin-3.5.8/kuser/globals.h.orig	2008-04-02 20:47:37.451050053 +0530
+++ kdeadmin-3.5.8/kuser/globals.h	2008-04-02 20:49:08.858601987 +0530
@@ -38,8 +38,13 @@
   #define MAIL_SPOOL_DIR _PATH_MAILDIR
 #else
   #define SHELL_FILE "/etc/shells"
+
+#if defined(USE_SOLARIS)
+  #define MAIL_SPOOL_DIR "/var/mail"
+#else
   #define MAIL_SPOOL_DIR "/var/spool/mail"
 #endif
+#endif
 
 #if defined(__FreeBSD__) || defined(__bsdi__)
   #undef HAVE_SHADOW
@@ -57,8 +62,12 @@
   #define GRMKDB "cd /var/yp; make 2>&1 >> /var/log/kuser"
   #define SKELDIR "/etc/skel"
   #define SKEL_FILE_PREFIX ""
+#if defined(USE_SOLARIS)
+  #define CRONTAB_DIR "/var/cron"
+#else
   #define CRONTAB_DIR "/var/spool/cron"
 #endif
+#endif
 
 #define GROUP_FILE "/etc/group"
 #define GROUP_FILE_MASK S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH
