diff -ur exo-0.3.2-orig/exo/exo-mount-point.c exo-0.3.2/exo/exo-mount-point.c
--- exo-0.3.2-orig/exo/exo-mount-point.c	2007-01-20 20:58:22.000000000 +0700
+++ exo-0.3.2/exo/exo-mount-point.c	2007-01-25 13:04:47.876867659 +0700
@@ -227,7 +227,7 @@
   FILE         *fp;
 
   /* try to open the /etc/mnttab file */
-  fp = setmntent ("/etc/mnttab", "r");
+  fp = fopen (_PATH_FSTAB, "r");
   if (G_UNLIKELY (fp == NULL))
     {
       g_set_error (error, G_FILE_ERROR, g_file_error_from_errno (errno),
