--- openmotif-2.3.0/lib/Xm/XmIm.c.orig	2008-02-20 06:09:03.066174728 -0800
+++ openmotif-2.3.0/lib/Xm/XmIm.c	2008-02-20 06:09:17.103791772 -0800
@@ -72,17 +72,6 @@
 # define Va_start(a,b) va_start(a,b)
 
 
-/*
-    On Sun XICProc is not defined. This should be dealt with
-    in the script: configure.ac however 1) not everyone uses that
-    2) AC_CHECK_TYPE(XICProc,,AC_DEFINE(...), include= ?) doesnt
-    work quite right/or is poorly documented and i cant get it to
-    work.
-*/
-#if defined(sun)
-typedef Bool (*XICProc)( XIC, XPointer, XPointer);
-#endif
-
 /* Data structures:
  *	While multiple XIMs are not currently supported, some thought
  * was given to how they might be implemented.  Currently both
