diff -ur thunar-archive-plugin-orig/thunar-archive-plugin/tap-backend.h thunar-archive-plugin/thunar-archive-plugin/tap-backend.h
--- thunar-archive-plugin-orig/thunar-archive-plugin/tap-backend.h	2007-04-19 14:07:32.000000000 +0700
+++ thunar-archive-plugin/thunar-archive-plugin/tap-backend.h	2007-04-19 14:15:20.710657671 +0700
@@ -23,23 +23,26 @@
 
 #include <thunarx/thunarx.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
+G_GNUC_INTERNAL
 GPid tap_backend_create_archive (const gchar *folder,
                                  GList       *files,
                                  GtkWidget   *window,
-                                 GError     **error) G_GNUC_INTERNAL;
+                                 GError     **error);
 
+G_GNUC_INTERNAL
 GPid tap_backend_extract_here   (const gchar *folder,
                                  GList       *files,
                                  GtkWidget   *window,
-                                 GError     **error) G_GNUC_INTERNAL;
+                                 GError     **error);
 
+G_GNUC_INTERNAL
 GPid tap_backend_extract_to     (const gchar *folder,
                                  GList       *files,
                                  GtkWidget   *window,
-                                 GError     **error) G_GNUC_INTERNAL;
+                                 GError     **error);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__TAP_BACKEND_H__ */
diff -ur thunar-archive-plugin-orig/thunar-archive-plugin/tap-provider.h thunar-archive-plugin/thunar-archive-plugin/tap-provider.h
--- thunar-archive-plugin-orig/thunar-archive-plugin/tap-provider.h	2007-04-19 14:07:32.000000000 +0700
+++ thunar-archive-plugin/thunar-archive-plugin/tap-provider.h	2007-04-19 14:16:01.718039571 +0700
@@ -23,7 +23,7 @@
 
 #include <thunarx/thunarx.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
 typedef struct _TapProviderClass TapProviderClass;
 typedef struct _TapProvider      TapProvider;
@@ -35,9 +35,11 @@
 #define TAP_IS_PROVIDER_CLASS(klass)  (G_TYPE_CHECK_CLASS_TYPE ((klass), TAP_TYPE_PROVIDER))
 #define TAP_PROVIDER_GET_CLASS(obj)   (G_TYPE_INSTANCE_GET_CLASS ((obj), TAP_TYPE_PROVIDER, TapProviderClass))
 
-GType tap_provider_get_type      (void) G_GNUC_CONST G_GNUC_INTERNAL;
-void  tap_provider_register_type (ThunarxProviderPlugin *plugin) G_GNUC_INTERNAL;
+G_GNUC_CONST G_GNUC_INTERNAL
+GType tap_provider_get_type      (void);
+G_GNUC_INTERNAL
+void  tap_provider_register_type (ThunarxProviderPlugin *plugin);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__TAP_PROVIDER_H__ */
