--- Terminal-0.2.8/terminal/terminal-gtk-extensions.h.fixgccism	2007-12-09 23:15:02.236907946 +0100
+++ Terminal-0.2.8/terminal/terminal-gtk-extensions.h	2007-12-09 23:16:19.735404829 +0100
@@ -22,15 +22,16 @@
 
 #include <gtk/gtk.h>
 
-G_BEGIN_DECLS;
+G_BEGIN_DECLS
 
-void terminal_gtk_label_set_a11y_relation (GtkLabel    *label,
-                                           GtkWidget   *widget) G_GNUC_INTERNAL;
+G_GNUC_INTERNAL void terminal_gtk_label_set_a11y_relation (GtkLabel    *label,
+                                           GtkWidget   *widget);
 
+G_GNUC_INTERNAL
 void terminal_gtk_widget_set_tooltip      (GtkWidget   *widget,
                                            const gchar *format,
-                                           ...) G_GNUC_INTERNAL G_GNUC_PRINTF (2, 3);
+                                           ...) G_GNUC_PRINTF (2, 3);
 
-G_END_DECLS;
+G_END_DECLS
 
 #endif /* !__TERMINAL_GTK_EXTENSIONS_H__ */
