--- xfce4-session-4.4.2/libxfsm/xfsm-util.c.orig	2008-03-31 13:24:42.895558791 +0530
+++ xfce4-session-4.4.2/libxfsm/xfsm-util.c	2008-03-31 13:31:54.017637299 +0530
@@ -237,8 +237,24 @@
 XfceRc*
 xfsm_open_config (gboolean readonly)
 {
-  return xfce_rc_config_open (XFCE_RESOURCE_CONFIG,
+  XfceRc *rc;
+  const gchar *rcenv;
+
+  /*
+   * Allow user to override session rc file under
+   * XFCE_RESOURCE_CONFIG.
+   */
+  rcenv = g_getenv("XFCE_SESSION_RC");
+  if (rcenv != NULL) {
+    rc = xfce_rc_config_open(XFCE_RESOURCE_CONFIG,
+                                    rcenv, readonly);
+  }
+
+  if (rcenv == NULL || rc == NULL)
+    rc = xfce_rc_config_open (XFCE_RESOURCE_CONFIG,
                               "xfce4-session/xfce4-session.rc",
                               readonly);
+
+  return (rc);
 }
 
--- xfce4-session-4.4.2/settings/splash/splash.c.orig	2008-03-31 14:54:35.156948473 +0530
+++ xfce4-session-4.4.2/settings/splash/splash.c	2008-03-31 15:05:31.099224978 +0530
@@ -222,11 +222,24 @@
 
   if (gtk_tree_selection_get_selected (selection, &model, &iter))
     {
+      const gchar *rcenv;
       gtk_tree_model_get (model, &iter, COLUMN_MODULE, &module, -1);
 
-      rc = xfce_rc_config_open (XFCE_RESOURCE_CONFIG,
-                                "xfce4-session/xfce4-session.rc",
-                                FALSE);
+      /*
+       * Allow user to override session rc file under
+       * XFCE_RESOURCE_CONFIG.
+       */
+      rcenv = g_getenv("XFCE_SESSION_RC");
+      if (rcenv != NULL) {
+        rc = xfce_rc_config_open(XFCE_RESOURCE_CONFIG,
+                                        rcenv, FALSE);
+      }
+
+      if (rcenv == NULL || rc == NULL)
+        rc = xfce_rc_config_open (XFCE_RESOURCE_CONFIG,
+                                  "xfce4-session/xfce4-session.rc",
+                                  FALSE);
+
       xfce_rc_set_group (rc, "Splash Screen");
 
       if (module != NULL)
--- xfce4-session-4.4.2/settings/session/session.c.orig	2008-03-31 14:55:08.269808401 +0530
+++ xfce4-session-4.4.2/settings/session/session.c	2008-03-31 15:06:09.164345890 +0530
@@ -68,9 +68,25 @@
 static XfceRc*
 config_open (gboolean readonly)
 {
-  return xfce_rc_config_open (XFCE_RESOURCE_CONFIG,
+  XfceRc *rc;
+  const gchar *rcenv;
+
+  /*
+   * Allow user to override session rc file under
+   * XFCE_RESOURCE_CONFIG.
+   */
+  rcenv = g_getenv("XFCE_SESSION_RC");
+  if (rcenv != NULL) {
+    rc = xfce_rc_config_open(XFCE_RESOURCE_CONFIG,
+                                    rcenv, readonly);
+  }
+
+  if (rcenv == NULL || rc == NULL)
+    rc = xfce_rc_config_open (XFCE_RESOURCE_CONFIG,
                               "xfce4-session/xfce4-session.rc",
                               readonly);
+
+  return (rc);
 }
 
 
