--- gnome-control-center-2.25.2/capplets/about-me/gnome-about-me.c.ori	2008-12-01 13:32:12.000000000 +0000
+++ gnome-control-center-2.25.2/capplets/about-me/gnome-about-me.c	2008-12-24 09:02:38.929614813 +0000
@@ -886,7 +886,25 @@
 	GladeXML *dialog;
 
 	dialog = me->dialog;
+#if (defined(sun) && defined(__SVR4))
+    /* Use gnome-terminal embedded passwd due to inability to fork passwd in
+     * the background.
+     */
+    {
+        GError  *error = NULL;
+        gchar   *command = NULL;
+
+        command = g_strdup_printf("gnome-terminal --window --title '%s' --hide-menubar -e /usr/bin/passwd",
+                                _("Change password") );
+
+        if ( command != NULL ) {
+            g_spawn_command_line_async( command, &error);
+            g_free(command);
+        }
+    }
+#else
 	gnome_about_me_password (GTK_WINDOW (WID ("about-me-dialog")));
+#endif
 }
 
 static void
