--- Displays/SideCandy/cpu.display.old	Sat Jan 22 06:26:15 2005
+++ Displays/SideCandy/cpu.display	Thu Feb 15 19:04:38 2007
@@ -119,7 +119,7 @@
     FADING_STEPS = 5
     DOM = Dsp.icon.dom
     cpu_top = DOM.get("Chip_Top")
-    refresh = 500
+    refresh = 5000
     # the fading context; the current color and the destination color
     fading_context = [(0, 0, 0), (0, 0, 0)]
 
@@ -129,10 +129,13 @@
         Dsp.lbl_cpu_info.visible = show
         Dsp.gauge_load.visible = Dsp.plot_load.visible = not show
         if (show):
-            Dsp.lbl_cpu_info.value = "Bogomips: %.2f\n%s (%dkB cache)" % \
+            if (sys.cpu_bogomips > 0):	
+                Dsp.lbl_cpu_info.value = "Bogomips: %.2f\n%s (%dkB cache)" % \
                               (sys.cpu_bogomips, sys.cpu_model, sys.cpu_cache)
+            else:
+                Dsp.lbl_cpu_info.value = "%s (%dkB cache)" % \
+                              (sys.cpu_model, sys.cpu_cache)
 
-
     def get_load():
 
         load = int(sys.cpu_load)
@@ -143,7 +146,7 @@
             clock = str("%4.f MHz" % sys.cpu_clock)
 
         if (len(sys.cpu_model) > 15): cpu_model = sys.cpu_model[:14] + ".."
-        else: cpu_model = sys.cpu_mode
+        else: cpu_model = sys.cpu_model
         Dsp.lbl_cpu.value = "%s @ %s" % (cpu_model, clock)
         Dsp.cpu_load.value = "%d%%" % load
         Dsp.plot_load.value = Dsp.gauge_load.fill = load
--- Displays/SideCandy/network.display.old	Sat Jan 22 06:35:47 2005
+++ Displays/SideCandy/network.display	Fri Feb 16 11:04:59 2007
@@ -124,8 +124,11 @@
   <script uri="candy.script"/>
 
   <script><![CDATA[
-
-    iface = "eth0"
+    iface = "lo0"
+    for i in sys.net_devices:
+        if i != "lo0":
+           iface = i	
+           break	
     
     def prefs_cb(key, value):
     
@@ -161,7 +164,7 @@
             return "%dB" % bytes
 
  
-    add_timer(500, get_load)
+    add_timer(5000, get_load)
     prefs_cb("iface", iface)
 
   ]]></script>
--- Displays/WeeklyCalendar/WeeklyCalendar.display.old	Tue Jan 10 08:55:08 2006
+++ Displays/WeeklyCalendar/WeeklyCalendar.display	Thu Feb 15 19:04:38 2007
@@ -485,7 +485,7 @@
             _m = 12
             _y -= 1
         draw_cal(_y, _m)
-        set_surrounding_months(_y, _m)
+       
     def cal_cur():
         y,m,d = time.date
         global _m, _y
@@ -493,7 +493,7 @@
         _m = m
         _y = y
         draw_cal()
-        set_surrounding_months(_y, _m)
+       
     def cal_next():
         y,m,d = time.date
         global _m, _y
@@ -505,7 +505,7 @@
             _y += 1
             _m = 1
         draw_cal(_y, _m)
-        set_surrounding_months(_y, _m)
+       
 
     #######################################
     # Things to do upon initialization
