--- gDesklets-0.36beta/main/Starter.py.old	2007-11-21 17:47:33.175854000 +0800
+++ gDesklets-0.36beta/main/Starter.py	2007-11-21 17:47:57.677739000 +0800
@@ -98,8 +98,8 @@
         # setup a nice systray icon
         if (settings.show_tray_icon):
             from main.TrayIcon import TrayIcon
-            trayicon = TrayIcon()
-            trayicon.set_menu([(None, _("_Manage desklets"),
+            self.__trayicon = TrayIcon()
+            self.__trayicon.set_menu([(None, _("_Manage desklets"),
                                 self.__handle_manage),
                                (),
                                (gtk.STOCK_PROPERTIES, _("_Configuration"),
