--- gtkhtml-3.26.0/configure.in.orig	Tue Apr  7 17:54:48 2009
+++ gtkhtml-3.26.0/configure.in	Tue Apr  7 17:55:22 2009
@@ -154,8 +154,8 @@
 
 if test "x$have_iso_codes" = "xyes"; then
 	AC_MSG_CHECKING([whether iso-codes has iso-639 and iso-3166 domains])
-	if $PKG_CONFIG --variable=domains iso-codes | grep -q 639 && \
-		$PKG_CONFIG --variable=domains iso-codes | grep -q 3166 ; then
+	if $PKG_CONFIG --variable=domains iso-codes | grep 639 && \
+		$PKG_CONFIG --variable=domains iso-codes | grep 3166 ; then
 			result=yes
 	else
 		result=no
