--- libgksu1.2-1.3.1.orig/libgksu/defines.h	2005-05-13 01:26:56.000000000 +0800
+++ libgksu1.2-1.3.1/libgksu/defines.h	2006-10-19 15:28:29.605168000 +0800
@@ -21,13 +21,6 @@
 #ifndef __DEFINES_H__
 #define __DEFINES_H__
 
-/* Gettext */
-#include <libintl.h>
-#define _(String) dgettext (PACKAGE, String)
-#define gettext_noop(String) (String)
-#ifndef HAVE_GNOME
-#define N_(String) gettext_noop (String)
-#endif
-/* Gettext */
+#include <glib/gi18n.h>
 
 #endif
