--- libgpg-error-1.5/configure.ac.orig	2007-10-01 22:42:30.873974000 -0400
+++ libgpg-error-1.5/configure.ac	2007-10-01 23:38:30.235678000 -0400
@@ -99,10 +99,12 @@
 #endif
 ])
 
-
-# Note, that autogen.sh greps for the next line.
-AM_GNU_GETTEXT_VERSION([0.15])
-AM_GNU_GETTEXT([external])
+LTLIBINTL=
+AC_SUBST(LTLIBINTL)
+GETTEXT_PACKAGE=libgpg-error
+AC_SUBST(GETTEXT_PACKAGE)
+ALL_LINGUAS="cs de es fr hu it ja ko pl pt_BR ro ru sv vi zh_CN zh_HK zh_TW"
+AM_GLIB_GNU_GETTEXT
 
 # Checks for header files.
 AC_HEADER_STDC
