--- kdebase-workspace-4.3.1/startkde.cmake.orig	2009-07-08 20:15:04.000000000 +0530
+++ kdebase-workspace-4.3.1/startkde.cmake	2009-09-05 23:38:51.547642509 +0530
@@ -1,8 +1,18 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #  DEFAULT KDE STARTUP SCRIPT ( @KDE_VERSION_STRING@ )
 #
 
+# Make sure we can find the binaries.
+PATH=@CMAKE_INSTALL_PREFIX@/qt4/bin:${PATH}
+export PATH
+
+# The .UTF-8 language settings cause iiimx to come up; this causes
+# plasma to crash - vanish - all the time with bad X_CopyArea
+# requests. So remove that setting because we have our own means
+# to change input methods.
+test -n "$LANG" && LANG=`basename "$LANG" ".UTF-8"`
+
 if test "x$1" = x--failsafe; then
     KDE_FAILSAFE=1 # General failsafe flag
     KWIN_COMPOSE=N # Disable KWin's compositing
