--- kcheckpass/checkpass_pam.c.orig	2008-05-07 11:05:23.000000000 +0200
+++ kcheckpass/checkpass_pam.c	2009-07-18 10:07:07.278635793 +0200
@@ -38,12 +38,12 @@
   int classic:1;
 };
 
-#ifdef PAM_MESSAGE_NONCONST
-typedef struct pam_message pam_message_type;
-typedef void *pam_gi_type;
-#else
+#ifdef PAM_MESSAGE_CONST
 typedef const struct pam_message pam_message_type;
 typedef const void *pam_gi_type;
+#else
+typedef struct pam_message pam_message_type;
+typedef void *pam_gi_type;
 #endif
 
 static int
