--- kdebindings-4.2.4/python/pykde4/sip/kio/kacl.sip.orig	2009-06-14 22:55:37.687852872 +0530
+++ kdebindings-4.2.4/python/pykde4/sip/kio/kacl.sip	2009-06-14 22:52:04.910792007 +0530
@@ -144,7 +144,8 @@
     {
         PyObject *elem = PyList_GET_ITEM (sipPy, i);
         PyObject *pyp1 = PyTuple_GET_ITEM (elem, 0);
-        p1 = *(QString *)sipForceConvertTo_QString (pyp1, &iserr);
+        p1 = *(QString *)sipForceConvertToType(pyp1, sipType_QString, \
+            sipTransferObj, SIP_NO_CONVERTORS, NULL, &iserr);
 
         if (iserr)
         {
