--- kdebindings-4.2.4/python/pykde4/sip/kio/global.sip.orig	2009-06-14 22:55:51.639918000 +0530
+++ kdebindings-4.2.4/python/pykde4/sip/kio/global.sip	2009-06-14 23:04:32.514068411 +0530
@@ -243,8 +243,10 @@
     {
         int iserr = 0;
 
-        acpp = (QString *)sipForceConvertTo_QString (aelem, &iserr);
-        bcpp = (QString *)sipForceConvertTo_QString (belem, &iserr);
+        acpp = (QString *)sipForceConvertToType(aelem, sipType_QString, sipTransferObj, \
+           SIP_NO_CONVERTORS, NULL, &iserr);
+        bcpp = (QString *)sipForceConvertToType(belem, sipType_QString, sipTransferObj, \
+           SIP_NO_CONVERTORS, NULL, &iserr);
 
         if (iserr)
         {
