--- evolution-2.26.0/mail/em-composer-utils.c.bak	Thu Apr  9 18:04:01 2009
+++ evolution-2.26.0/mail/em-composer-utils.c	Thu Apr  9 18:06:53 2009
@@ -1859,7 +1859,22 @@
 	    && (account = guess_account_folder(folder)))
 		return account;
 
-	/* then recipient (to/cc) in account table */
+	/* check for source folder */
+	if (folder) {
+		account = guess_account_folder(folder);
+		if (account)
+			return account;
+	}
+
+	/* then message source */
+	if (account == NULL
+	    && (tmp = camel_mime_message_get_source(message))) {
+		account = mail_config_get_account_by_source_url(tmp);
+		if (account)
+			return account;
+	}
+
+	/* finally recipient (to/cc) in account table */
 	account_hash = generate_account_hash ();
 	for (j=0;account == NULL && j<2;j++) {
 		const CamelInternetAddress *to;
@@ -1875,16 +1890,6 @@
 	}
 	g_hash_table_destroy(account_hash);
 
-	/* then message source */
-	if (account == NULL
-	    && (tmp = camel_mime_message_get_source(message)))
-		account = mail_config_get_account_by_source_url(tmp);
-
-	/* and finally, source folder */
-	if (account == NULL
-	    && folder)
-		account = guess_account_folder(folder);
-
 	return account;
 }
 
