--- gdm-2.20.9/config/locale.alias.orig	2009-03-04 14:31:17.550354000 +0900
+++ gdm-2.20.9/config/locale.alias	2009-03-04 15:54:59.945754000 +0900
@@ -60,6 +60,7 @@ French(Canada)		fr_CA.UTF-8,fr_CA
 French(Luxembourg)	fr_LU.UTF-8,fr_LU
 French(Switzerland)	fr_CH.UTF-8,fr_CH
 Galician		gl_ES.UTF-8,gl_ES
+Georgian		ka_GE.UTF-8,ka_GE
 German			de_DE.UTF-8,de_DE
 German(Austria)		de_AT.UTF-8,de_AT
 German(Switzerland)	de_CH.UTF-8,de_CH
@@ -126,6 +127,7 @@ Spanish(Panama)		es_PA.UTF-8,es_PA
 Spanish(Peru)		es_PE.UTF-8,es_PE
 Spanish(Paraguay)	es_PY.UTF-8,es_PY
 Spanish(ElSalvador)	es_SV.UTF-8,es_SV
+Spanish(USA)		es_US.UTF-8,es_US
 Spanish(Uruguay)	es_UY.UTF-8,es_UY
 Spanish(Venezuela)	es_VE.UTF-8,es_VE
 Swedish(Finland)	sv_FI.UTF-8,sv_FI
--- gdm-2.20.1/gui/gdmlanguages.c.orig	2009-03-04 15:55:26.552931000 +0900
+++ gdm-2.20.1/gui/gdmlanguages.c	2009-03-04 17:29:47.184533000 +0900
@@ -161,6 +161,9 @@ static Language languages [] = {
 	/*Note translate the A-M to the A-M you used in the group label */
 	{ N_("A-M|Galician"), "gl_ES", "Galego", 0 },
 	/*Note translate the A-M to the A-M you used in the group label */
+	/* SUN_BRANDING */
+	{ N_("A-M|Georgian"), "ka_GE", NULL, 0 },
+	/*Note translate the A-M to the A-M you used in the group label */
 	{ N_("A-M|German"), "de_DE", "Deutsch", 0 },
 	/*Note translate the A-M to the A-M you used in the group label */
 	{ N_("A-M|German (Austria)"), "de_AT", "Deutsch (Österreich)", 0 },
@@ -183,7 +186,7 @@ static Language languages [] = {
 	/*Note translate the A-M to the A-M you used in the group label */
 	{ N_("A-M|Hungarian"), "hu_HU", "Magyar", 0 },
 	/*Note translate the A-M to the A-M you used in the group label */
-	{ N_("A-M|Icelandic"), "is_IS", NULL, 0 },
+	{ N_("A-M|Icelandic"), "is_IS", "íslenska", 0 },
 	/*Note translate the A-M to the A-M you used in the group label */
 	{ N_("A-M|Indonesian"), "id_ID", "Bahasa Indonesia", 0 },
 	/*Note translate the A-M to the A-M you used in the group label */
@@ -195,6 +198,9 @@ static Language languages [] = {
 	/*Note translate the A-M to the A-M you used in the group label */
 	{ N_("A-M|Japanese"), "ja_JP", "日本語", 0 },
 	/*Note translate the A-M to the A-M you used in the group label */
+	/* SUN_BRANDING */
+	{ N_("A-M|Kazakh"), "kk_KZ", "Qazaq", 0 },
+	/*Note translate the A-M to the A-M you used in the group label */
 	{ N_("A-M|Kannada"), "kn_IN", "﻿ಕನ್ನಡ", 0 },
 	/*Note translate the A-M to the A-M you used in the group label */
 	{ N_("A-M|Kinyarwanda"), "rw_RW", "Kinyarwanda", 0 },
@@ -219,6 +225,8 @@ static Language languages [] = {
 	/*Note translate the N-Z to the N-Z you used in the group label */
 	{ N_("N-Z|Northern Sotho"), "nso_ZA", "Sesotho sa Leboa", 0 },
 	/*Note translate the N-Z to the N-Z you used in the group label */
+	{ N_("N-Z|Norwegian (bokmal)"), "nb_NO", "Norsk (bokmål)", 0 },
+	/*Note translate the N-Z to the N-Z you used in the group label */
 	{ N_("N-Z|Norwegian (bokmal)"), "no_NO", "Norsk (bokmål)", 0 },
 	/*Note translate the N-Z to the N-Z you used in the group label */
 	{ N_("N-Z|Norwegian (nynorsk)"), "nn_NO", "Norsk (nynorsk)", 0 },
@@ -285,6 +293,9 @@ static Language languages [] = {
 	/*Note translate the N-Z to the N-Z you used in the group label */
 	{ N_("N-Z|Spanish (El Salvador)"), "es_SV", "Español (El Salvador)", 0 },
 	/*Note translate the N-Z to the N-Z you used in the group label */
+	/* SUN_BRANDING */
+	{ N_("N-Z|Spanish (USA)"), "es_US", "Español (USA)", 0 },
+	/*Note translate the N-Z to the N-Z you used in the group label */
 	{ N_("N-Z|Spanish (Uruguay)"), "es_UY", "Español (Uruguay)", 0 },
 	/*Note translate the N-Z to the N-Z you used in the group label */
 	{ N_("N-Z|Spanish (Venezuela)"), "es_VE", "Español (Venezuela)", 0 },
@@ -493,7 +504,12 @@ gdm_lang_untranslated_name (const char *
 	if (lang == NULL)
 		return NULL;
 
-	return g_strdup (lang->untranslated);
+	if (lang->untranslated) {
+		return g_strdup (lang->untranslated);
+	} else {
+		char *p = strchr (lang->name, '|');
+		return g_strdup (p ? (p + 1) : lang->name);
+	}
 }
 
 const char *
