--- ./src/ptlib/common/sockets.cxx.orig	2009-03-22 15:02:37.903419000 +0800
+++ ./src/ptlib/common/sockets.cxx	2009-03-22 15:03:38.175157000 +0800
@@ -535,7 +535,8 @@
     struct addrinfo hints = { AI_CANONNAME, defaultIpAddressFamily };
     localErrNo = getaddrinfo((const char *)name, NULL , &hints, &res);
     host = new PIPCacheData(localErrNo != NETDB_SUCCESS ? NULL : res, name);
-    freeaddrinfo(res);
+    if (res != NULL)
+      freeaddrinfo(res);
 
 #else // HAS_GETADDRINFO
 
@@ -1925,7 +1926,8 @@
         struct sockaddr_in * addr_in = (struct sockaddr_in *)res->ai_addr;
         v.four = addr_in->sin_addr;
       }
-      freeaddrinfo(res);
+      if (res != NULL)
+        freeaddrinfo(res);
       return IsValid();
     }
 
