--- usr/lib/gcc/i386-pc-solaris2.11/4.5.2/include-fixed/arpa/nameser_compat.h.orig	2009-05-19 21:45:18.170331634 +0530
+++ usr/lib/gcc/i386-pc-solaris2.11/4.5.2/include-fixed/arpa/nameser_compat.h	2009-05-19 21:45:53.297976467 +0530
@@ -127,7 +127,7 @@
 	 * which will force your compiles to bomb until you fix
 	 * the above macros.
 	 */
-	error "Undefined or invalid BYTE_ORDER";
+#        error "Undefined or invalid BYTE_ORDER";
 #endif
 
 /*
